--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1316,7 +1316,7 @@
                 this.stats.incConflicts();
                 this.slistener.conflictFound(confl, decisionLevel(),
                         this.trail.size());
-                this.conflictCount.newConflict();
+                
 
                 if (decisionLevel() == this.rootLevel) {
                     if (this.lastConflictMeansUnsat) {
@@ -1929,10 +1929,7 @@
         out.print(prefix);
         out.println("constraints type ");
         long total = 0;
-        for (Map.Entry<String, Counter> entry : this.constrTypes.entrySet()) {
-            out.println(prefix + entry.getKey() + " => " + entry.getValue());
-            total += entry.getValue().getValue();
-        }
+        
         out.print(prefix);
         out.print(total);
         out.println(" constraints processed.");
@@ -2479,6 +2476,7 @@
 
     public void setKeepSolverHot(boolean keepHot) {
         this.keepHot = keepHot;
+        this.timeBasedTimeout = false;
     }
 
     private final Comparator<Integer> trailComparator() {
