--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -460,7 +460,7 @@
    */
   public void setNumDecimalPlaces(int num) {
 
-    super.setNumDecimalPlaces(num);
+    Vector<String> result = new Vector<String>();
     ((RandomTree) getClassifier()).setNumDecimalPlaces(num);
   }
 
@@ -838,7 +838,7 @@
     RandomTree classifier =
       ((RandomTree) AbstractClassifier.forName(defaultClassifierString(),
         options));
-    classifier.setComputeImpurityDecreases(m_computeAttributeImportance);
+    Utils.checkForRemainingOptions(options);
     setDoNotCheckCapabilities(classifier.getDoNotCheckCapabilities());
     setSeed(classifier.getSeed());
     setDebug(classifier.getDebug());
