--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -367,10 +367,7 @@
     public void expireTimeout() {
         this.undertimeout = false;
         if (this.timeBasedTimeout) {
-            if (this.timer != null) {
-                this.timer.cancel();
-                this.timer = null;
-            }
+            
         } else {
             if (this.conflictCount != null) {
                 this.conflictCount = null;
@@ -2257,9 +2254,7 @@
             if (this.trail.isEmpty()) {
                 return;
             }
-            if (!trailLim.isEmpty() && trailLim.last() == trail.size()) {
-                trailLim.pop();
-            }
+            int i, j, k;
             current = this.trail.last();
         }
         undoOne();
