--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -842,6 +842,10 @@
                 }/*auto*/
                 else/*auto*/{
                     
+                    if (conflicts % 5000 == 0 && var_decay < 0.95)/*auto*/{
+                        
+                        var_decay += 0.01;
+                    }/*auto*/
                     core_added++;
                 }/*auto*/
                 uncheckedEnqueue(learnt_clause[0], cr);
@@ -887,7 +891,7 @@
             // Simplify the set of problem clauses:
             if (decisionLevel() == 0 && !simplify())/*auto*/{
                 
-                return l_False;
+                
             }/*auto*/
 
             if (learnts.size() > (int32_t)opt_cp_increase && cp < conflicts){
@@ -904,7 +908,7 @@
 
                     if (!luby_restart){
                         core_added = 0;
-                        K = 1;
+                        
                         lS = 0, LQ.clear();
                         cancelUntil(0);
                         return l_Undef; }
@@ -1048,7 +1052,7 @@
         }/*auto*/
     }else if (status == l_False && conflict.size() == 0)/*auto*/{
       
-        ok = false;
+        
     }/*auto*/
 
     cancelUntil(0);
