--- before: triangle.py
+++ after: triangle.py
@@ -6,10 +6,34 @@
     INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
 
 def delay():
-    time.sleep(0.001)
+    def classify_triangle(a, b, c):
+        delay()
+        if a > b:
+            tmp = a
+            a = b
+            b = tmp
+        if a > c:
+            tmp = a
+            a = c
+            c = tmp
+        if b > c:
+            tmp = b
+            b = c
+            c = tmp
+        if a + b <= c:
+            return TriangleType.INVALID
+        else:
+            if a == b and b == c:
+                return TriangleType.EQUILATERAL
+            else:
+                if a == b or b == c:
+                    return TriangleType.ISOSCELES
+                else:
+                    return TriangleType.SCALENE
 
 def classify_triangle(a, b, c):
-    delay()
+    
+    tmp = a
     if a > b:
         tmp = a
         a = b
@@ -21,6 +45,10 @@
     if b > c:
         tmp = b
         b = c
+        if a > c:
+            tmp = a
+            a = c
+            c = tmp
         c = tmp
     if a + b <= c:
         return TriangleType.INVALID
