--- before: minisat_simplified.params
+++ after: minisat_simplified.params
@@ -1,9 +1,9 @@
-luby := 'True'
+luby := 'False'
 rnd-init := 'False'
 gc-frac := 0.2
-rinc := 2.0
+rinc := 25424.441925227544
 var-decay := 0.95
-cla-decay := 0.999
+cla-decay := 0.8451034491307339
 rnd-freq := 0.0
 rnd-seed := 91648253
 phase-saving := 2
@@ -14,7 +14,7 @@
 rcheck := 'False'
 asymm := 'False'
 elim := 'True'
-simp-gc-frac := 0.5
-sub-lim := 1000
+simp-gc-frac := 218632158.1916924
+sub-lim := 65020
 cl-lim := 20
 grow := 0
