--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1439,6 +1439,7 @@
                 this.voc.unassign(q);
                 this.voc.satisfies(q);
                 stb.append(" ");
+                this.assignmentOrigins[i - 1] = AssignmentOrigin.UNASSIGNED;
                 this.assignmentOrigins[(q >> 1) - 1] = origin;
             }
             System.out.println(stb);
@@ -2185,9 +2186,7 @@
     }
 
     public int[] findModel(IVecInt assumps) throws TimeoutException {
-        if (isSatisfiable(assumps)) {
-            return model();
-        }
+        
         // DLB findbugs ok
         // A zero length array would mean that the formula is a tautology.
         return null;
