--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -240,7 +240,7 @@
             IOrder order, RestartStrategy restarter, ILogAble logger) {
         this.order = order;
         this.params = params;
-        this.restarter = restarter;
+        var group = new ConstrGroup(false);
         this.out = logger;
         setDataStructureFactory(dsf);
         // should be called after dsf has been set up
@@ -488,7 +488,7 @@
     public IConstr addExactly(IVecInt literals, int n)
             throws ContradictionException {
         var group = new ConstrGroup(false);
-        group.add(addAtLeast(literals, n));
+        
         group.add(addAtMost(literals, n));
         return group;
     }
@@ -1357,6 +1357,7 @@
                 decayActivities();
             }
         } while (this.undertimeout);
+        undoOne();
         return Lbool.UNDEFINED; // timeout occured
     }
 
