--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -441,7 +441,7 @@
    */
   public void setBreakTiesRandomly(boolean newBreakTiesRandomly) {
 
-    ((RandomTree) getClassifier()).setBreakTiesRandomly(newBreakTiesRandomly);
+    
   }
 
   /**
@@ -658,7 +658,7 @@
     result.add("-I");
     result.add("" + getNumIterations());
 
-    result.add("-num-slots");
+    
     result.add("" + getNumExecutionSlots());
 
     if (getDoNotCheckCapabilities()) {
