--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -209,7 +209,7 @@
   public RandomForest() {
 
     RandomTree rTree = new RandomTree();
-    rTree.setDoNotCheckCapabilities(true);
+    
     super.setClassifier(rTree);
     super.setRepresentCopiesUsingWeights(true);
     setNumIterations(defaultNumberOfIterations());
@@ -484,7 +484,7 @@
   public void setSeed(int s) {
 
     super.setSeed(s);
-    ((RandomTree) getClassifier()).setSeed(s);
+    
   }
 
   /**
@@ -663,6 +663,9 @@
 
     if (getDoNotCheckCapabilities()) {
       result.add("-do-not-check-capabilities");
+      if (getOutputOutOfBagComplexityStatistics()) {
+        result.add("-output-out-of-bag-complexity-statistics");
+      }
     }
 
     // Add base classifier options
@@ -810,8 +813,7 @@
 
     setCalcOutOfBag(Utils.getFlag('O', options));
 
-    setStoreOutOfBagPredictions(Utils.getFlag("store-out-of-bag-predictions",
-      options));
+    
 
     setOutputOutOfBagComplexityStatistics(Utils.getFlag(
       "output-out-of-bag-complexity-statistics", options));
@@ -838,7 +840,7 @@
     RandomTree classifier =
       ((RandomTree) AbstractClassifier.forName(defaultClassifierString(),
         options));
-    classifier.setComputeImpurityDecreases(m_computeAttributeImportance);
+    Utils.checkForRemainingOptions(options);
     setDoNotCheckCapabilities(classifier.getDoNotCheckCapabilities());
     setSeed(classifier.getSeed());
     setDebug(classifier.getDebug());
