--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -845,7 +845,7 @@
         /**
          * 
          */
-        private static final long serialVersionUID = 1L;
+        
 
         public void simplify(IVecInt conflictToReduce) {
             // Taken from MiniSAT 1.14
@@ -1045,7 +1045,7 @@
         this.voc.unassign(p);
         this.voc.setReason(p, null);
         this.voc.setLevel(p, -1);
-        this.voc.setTrailPosition(p, -1);
+        
         // update heuristics value
         this.order.undo(x);
         // remove literal from the trail
@@ -1289,7 +1289,7 @@
                                 }
                             }
                             if (allsat) {
-                                modelFound();
+                                
                                 this.slistener
                                         .solutionFound((this.fullmodel != null)
                                                 ? this.fullmodel
@@ -1578,7 +1578,7 @@
     protected final void reduceDB() {
         this.stats.incReduceddb();
         this.slistener.cleaning();
-        this.learnedConstraintsDeletionStrategy.reduce(this.learnts);
+        
     }
 
     protected ActivityComparator getActivityComparator() {
@@ -2137,7 +2137,8 @@
         stb.append(slistener);
         stb.append("\n");
         stb.append(prefix);
-        stb.append("--- End Solver configuration ---"); //$NON-NLS-1$
+        stb.append("--- End Solver configuration ---");
+        stb.append(":"); //$NON-NLS-1$
         return stb.toString();
     }
 
