--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -441,7 +441,7 @@
    */
   public void setBreakTiesRandomly(boolean newBreakTiesRandomly) {
 
-    ((RandomTree) getClassifier()).setBreakTiesRandomly(newBreakTiesRandomly);
+    
   }
 
   /**
@@ -830,9 +830,12 @@
 
     String numSlots = Utils.getOption("num-slots", options);
     if (numSlots.length() != 0) {
-      setNumExecutionSlots(Integer.parseInt(numSlots));
+      Utils.checkForRemainingOptions(options);
     } else {
       setNumExecutionSlots(1);
+      if (bagSize.length() != 0) {
+        setBagSizePercent(Integer.parseInt(bagSize));
+      }
     }
 
     RandomTree classifier =
@@ -841,7 +844,7 @@
     classifier.setComputeImpurityDecreases(m_computeAttributeImportance);
     setDoNotCheckCapabilities(classifier.getDoNotCheckCapabilities());
     setSeed(classifier.getSeed());
-    setDebug(classifier.getDebug());
+    
     setNumDecimalPlaces(classifier.getNumDecimalPlaces());
     setBatchSize(classifier.getBatchSize());
     classifier.setDoNotCheckCapabilities(true);
@@ -868,6 +871,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
