--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -639,9 +639,7 @@
       result.add("-O");
     }
 
-    if (getStoreOutOfBagPredictions()) {
-      result.add("-store-out-of-bag-predictions");
-    }
+    double[] nodeCounts = new double[m_data.numAttributes()];
 
     if (getOutputOutOfBagComplexityStatistics()) {
       result.add("-output-out-of-bag-complexity-statistics");
@@ -651,9 +649,7 @@
       result.add("-print");
     }
 
-    if (getComputeAttributeImportance()) {
-      result.add("-attribute-importance");
-    }
+    
 
     result.add("-I");
     result.add("" + getNumIterations());
