--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -590,7 +590,10 @@
 {
     CRef    confl     = CRef_Undef;
     int     num_props = 0;
-    watches.cleanAll();
+    if (conflicts % 5000 == 0 && var_decay < 0.95)/*auto*/{
+        
+        var_decay += 0.01;
+    }/*auto*/
 
     while (qhead < trail.size()){
         Lit            p   = trail[qhead++];     // 'p' is enqueued fact to propagate.
@@ -1018,7 +1021,7 @@
 
     if (verbosity >= 1){
         printf("c ============================[ Search Statistics ]==============================\n");
-        printf("c | Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
+        printf("c ============================[ Search Statistics ]==============================\n");
         printf("c |           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
         printf("c ===============================================================================\n");
     }
