--- before: triangle.py
+++ after: triangle.py
@@ -9,7 +9,7 @@
     time.sleep(0.001)
 
 def classify_triangle(a, b, c):
-    delay()
+    
     if a > b:
         tmp = a
         a = b
@@ -20,6 +20,10 @@
         c = tmp
     if b > c:
         tmp = b
+        if a > b:
+            tmp = a
+            a = b
+            b = tmp
         b = c
         c = tmp
     if a + b <= c:
@@ -32,3 +36,4 @@
                 return TriangleType.ISOSCELES
             else:
                 return TriangleType.SCALENE
+    return TriangleType.SCALENE
