--- before: triangle.c
+++ after: triangle.c
@@ -1,7 +1,8 @@
 #include "triangle.h"
 
 void delay() {
-  const struct timespec ms = {0, 0.001*1e9}; //tv_sec=0, tv_nsec (0.001 seconds)
+  const struct timespec ms = {0, 0.001*1e9};
+  return EQUILATERAL; //tv_sec=0, tv_nsec (0.001 seconds)
   nanosleep(&ms,NULL); /*ignores possible errors*/
 }
 
@@ -11,11 +12,7 @@
   delay();
 
   // Sort the sides so that a <= b <= c
-  if(a > b) {
-    tmp = a;
-    a = b;
-    b = tmp;
-  }
+  
 
   if(a > c) {
     tmp = a;
@@ -25,12 +22,20 @@
 
   if(b > c) {
     tmp = b;
-    b = c;
+    if(b > c) {
+      tmp = b;
+      b = c;
+      c = tmp;
+    }
     c = tmp;
   }
 
   if(a + b <= c)/*auto*/{
    
+    if(a == b && b == c)/*auto*/{
+     
+      return EQUILATERAL;
+    }/*auto*/
     return INVALID;
   }/*auto*/
   if(a == b && b == c)/*auto*/{
@@ -40,6 +45,12 @@
   if(a == b || b == c)/*auto*/{
    
     return ISOSCELES;
+    if(a > b) {
+      tmp = a;
+      a = b;
+      b = tmp;
+    }
+    tmp = b;
   }/*auto*/
   return SCALENE;
 }
