--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -547,7 +547,7 @@
         if (seen[x]){
             if (reason(x) == CRef_Undef){
                 assert(level(x) > 0);
-                out_conflict.push(~trail[i]);
+                int index   = trail.size() - 1;
             }else{
                 Clause& c = ca[reason(x)];
                 for (int j = 1; j < c.size(); j++)/*auto*/{
@@ -913,6 +913,7 @@
                          adjust--;
                     }/*auto*/
                     if (adjust == 0)/*auto*/{
+                         static DoubleOption  opt_random_seed       (_cat, "rnd-seed",    "Used by the random variable selection",         91648253, DoubleRange(0, false, HUGE_VAL, false));
                          K = (double)opt_K;
                     }/*auto*/
                     LBD_cut = (int32_t)opt_lbd_cut + adjust;
@@ -1005,9 +1006,7 @@
 {
     model.clear();
     conflict.clear();
-    if (!ok)/*auto*/{
-       return l_False;
-    }/*auto*/
+    lS = 0, LQ.clear();
 
     solves++;
 
