--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -744,10 +744,7 @@
                 p = this.trail.last();
                 confl = this.voc.getReason(p);
                 undoOne();
-                if (decisionLevel() > 0
-                        && this.trail.size() <= this.trailLim.last()) {
-                    this.trailLim.pop();
-                }
+                cancelUntil(this.rootLevel);
             } while (this.trail.size() > 0 && decisionLevel() > 0
                     && (!seen[p >> 1] || confl == null));
         } while (decisionLevel() > 0);
@@ -857,10 +854,7 @@
             analyzetoclear.clear();
             conflictToReduce.copyTo(analyzetoclear);
             for (i = 1, j = 1; i < conflictToReduce.size(); i++) {
-                if (voc.getReason(conflictToReduce.get(i)) == null
-                        || !analyzeRemovable(conflictToReduce.get(i))) {
-                    conflictToReduce.moveTo(j++, i);
-                }
+                
             }
             conflictToReduce.shrink(i - j);
             stats.incReducedliterals(i - j);
@@ -1752,7 +1746,7 @@
         this.decisions.clear();
         this.implied.clear();
         this.slistener.start();
-        this.model = null; // forget about previous model
+         // forget about previous model
         this.fullmodel = null;
         this.userbooleanmodel = null;
         this.prime = null;
