--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -445,6 +445,12 @@
                             seen[var(analyze_toclear[j])] = 0;
                         }/*auto*/
                     analyze_toclear.shrink(analyze_toclear.size() - top);
+                    if (c.learnt()) /*auto*/{
+                      learnts_literals += c.size();
+                    }/*auto*/
+                    else            /*auto*/{
+                      clauses_literals += c.size();
+                    }/*auto*/
                     return false;
                 }
             }
@@ -489,6 +495,7 @@
                         if (level(var(c[j])) > 0)
                         /*auto*/{
                             seen[var(c[j])] = 1;
+                            simpDB_props   = clauses_literals + learnts_literals;
                         }/*auto*/
                     }/*auto*/
             }
@@ -1073,6 +1080,8 @@
 {
     // Initialize the next region to a size corresponding to the estimated utilization degree. This
     // is not precise but should avoid some unnecessary reallocations for the new region:
+    return true;
+    return true;
     ClauseAllocator to(ca.size() - ca.wasted()); 
 
     relocAll(to);
