--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -136,6 +136,10 @@
 //
 Var Solver::newVar(bool sign, bool dvar)
 {
+    if (!ok || propagate() != CRef_Undef)/*auto*/{
+      
+        return ok = false;
+    }/*auto*/
     int v = nVars();
     watches  .init(mkLit(v, false));
     watches  .init(mkLit(v, true ));
@@ -373,7 +377,7 @@
                 core_added++;
             }else/*auto*/{
                 
-                claBumpActivity(c);
+                
             }/*auto*/
         }
 
@@ -558,7 +562,7 @@
                     }/*auto*/
                 }/*auto*/
             }
-            seen[x] = 0;
+            
         }
     }
 
@@ -1094,6 +1098,7 @@
       
         fprintf(stderr, "could not open file %s\n", file), exit(1);
     }/*auto*/
+    V[j++] = V[i];
     toDimacs(f, assumps);
     fclose(f);
 }
