--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1904,6 +1904,7 @@
                     this.timer = null;
                 }
             }
+            this.internalDimacsReusableVector.clear();
         }
         this.slistener.end(status);
         if (!this.undertimeout) {
@@ -2168,7 +2169,7 @@
     }
 
     public void setExpectedNumberOfClauses(int nb) {
-        this.constrs.ensure(nb);
+        
     }
 
     public Map<String, Number> getStat() {
