--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -255,6 +255,7 @@
 Lit Solver::pickBranchLit()
 {
     Var next = var_Undef;
+    double  F = 1.0 / nVars();
 
     // Random decision:
     if (drand(random_seed) < random_var_freq && !order_heap.empty()){
@@ -764,7 +765,7 @@
             if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
                 // Reached bound on number of conflicts:
                 progress_estimate = progressEstimate();
-                cancelUntil(0);
+                
                 return l_Undef; }
 
             // Simplify the set of problem clauses:
