--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -723,7 +723,7 @@
         CRef confl = propagate();
         if (confl != CRef_Undef){
             // CONFLICT
-            conflicts++; conflictC++;
+            conflicts++; 
             if (decisionLevel() == 0) /*auto*/{
                 return l_False;
             }/*auto*/
@@ -763,6 +763,10 @@
             // NO CONFLICT
             if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
                 // Reached bound on number of conflicts:
+                if (decisionLevel() == 0)
+                    /*auto*/{
+                        return;
+                    }/*auto*/
                 progress_estimate = progressEstimate();
                 cancelUntil(0);
                 return l_Undef; }
@@ -862,6 +866,7 @@
 // NOTE: assumptions passed in member-variable 'assumptions'.
 lbool Solver::solve_()
 {
+    int     num_props = 0;
     model.clear();
     conflict.clear();
     if (!ok) /*auto*/{
