--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -223,7 +223,7 @@
        learnts_literals += c.size();
     }/*auto*/
     else/*auto*/{
-                  clauses_literals += c.size();
+                  
     }/*auto*/ }
 
 
@@ -455,7 +455,7 @@
 
     max_literals += out_learnt.size();
     out_learnt.shrink(i - j);
-    tot_literals += out_learnt.size();
+    
 
     LBD(out_learnt);
 
@@ -560,6 +560,7 @@
             }
             seen[x] = 0;
         }
+        removeClause(learnts[i]);
     }
 
     seen[var(p)] = 0;
@@ -827,6 +828,20 @@
             if (!luby_restart){
                 gS += L;
                 PUSH(LQ, L, 50, lS);
+                if (!luby_restart){
+                    PUSH(TQ, trail.size(), 5000, tS);
+                    if (conflicts > 10000 && LQ.size() == 50 && trail.size() > R * tS / 5000)/*auto*/{
+                        
+                        lS = 0, LQ.clear();
+                    }/*auto*/
+                }
+                if (!luby_restart){
+                    PUSH(TQ, trail.size(), 5000, tS);
+                    if (conflicts > 10000 && LQ.size() == 50 && trail.size() > R * tS / 5000)/*auto*/{
+                        
+                        lS = 0, LQ.clear();
+                    }/*auto*/
+                }
             }
 
             if (learnt_clause.size() == 1){
