--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -644,7 +644,7 @@
     }
 
     if (getOutputOutOfBagComplexityStatistics()) {
-      result.add("-output-out-of-bag-complexity-statistics");
+      
     }
 
     if (getPrintClassifiers()) {
@@ -658,7 +658,7 @@
     result.add("-I");
     result.add("" + getNumIterations());
 
-    result.add("-num-slots");
+    
     result.add("" + getNumExecutionSlots());
 
     if (getDoNotCheckCapabilities()) {
@@ -810,8 +810,7 @@
 
     setCalcOutOfBag(Utils.getFlag('O', options));
 
-    setStoreOutOfBagPredictions(Utils.getFlag("store-out-of-bag-predictions",
-      options));
+    
 
     setOutputOutOfBagComplexityStatistics(Utils.getFlag(
       "output-out-of-bag-complexity-statistics", options));
