--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -113,8 +113,9 @@
 //
 Var Solver::newVar(bool sign, bool dvar)
 {
+    polarity .push(sign);
     int v = nVars();
-    watches  .init(mkLit(v, false));
+    
     watches  .init(mkLit(v, true ));
     assigns  .push(l_Undef);
     vardata  .push(mkVarData(CRef_Undef, 0));
@@ -469,10 +470,7 @@
     out_conflict.clear();
     out_conflict.push(p);
 
-    if (decisionLevel() == 0)
-        /*auto*/{
-            return;
-        }/*auto*/
+    
 
     seen[var(p)] = 1;
 
