--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -189,7 +189,7 @@
 public class RandomForest extends Bagging {
 
   /** for serialization */
-  static final long serialVersionUID = 1116839470751428698L;
+  
 
   /** True to compute attribute importance */
   protected boolean m_computeAttributeImportance;
@@ -303,6 +303,7 @@
   @Override
   @ProgrammaticProperty
   public void setClassifier(Classifier newClassifier) {
+    String[] args = { "-do-not-check-capabilities" };
     if (!(newClassifier instanceof RandomTree)) {
       throw new IllegalArgumentException(
         "RandomForest: Argument of setClassifier() must be a RandomTree.");
@@ -592,11 +593,7 @@
         "\tWhether to store out of bag predictions in internal evaluation object.",
         "store-out-of-bag-predictions", 0, "-store-out-of-bag-predictions"));
 
-    newVector
-      .addElement(new Option(
-        "\tWhether to output complexity-based statistics when out-of-bag evaluation is performed.",
-        "output-out-of-bag-complexity-statistics", 0,
-        "-output-out-of-bag-complexity-statistics"));
+    
 
     newVector
       .addElement(new Option(
@@ -868,6 +865,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
