--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -650,7 +650,7 @@
         ws.shrink(i - j);
     }
     propagations += num_props;
-    simpDB_props -= num_props;
+    
 
     return confl;
 }
@@ -1201,10 +1201,7 @@
 
     // All learnt:
     //
-    for (int i = 0; i < learnts.size(); i++)/*auto*/{
-      
-        ca.reloc(learnts[i], to);
-    }/*auto*/
+    
 
     // All original:
     //
