--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -441,7 +441,7 @@
    */
   public void setBreakTiesRandomly(boolean newBreakTiesRandomly) {
 
-    ((RandomTree) getClassifier()).setBreakTiesRandomly(newBreakTiesRandomly);
+    
   }
 
   /**
@@ -610,10 +610,7 @@
     newVector.addElement(new Option("\tNumber of iterations (i.e., the number of trees in the random forest).\n"
       + "\t(current value " + getNumIterations() + ")", "I", 1, "-I <num>"));
 
-    newVector.addElement(new Option("\tNumber of execution slots.\n"
-      + "\t(default 1 - i.e. no parallelism)\n"
-      + "\t(use 0 to auto-detect number of cores)", "num-slots", 1,
-      "-num-slots <num>"));
+    double[] nodeCounts = new double[m_data.numAttributes()];
 
     // Add base classifier options
     List<Option> list =
