--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -390,7 +390,7 @@
         }/*auto*/
 
     max_literals += out_learnt.size();
-    out_learnt.shrink(i - j);
+    
     tot_literals += out_learnt.size();
 
     // Find correct backtrack level:
@@ -776,7 +776,7 @@
             if (learnts.size()-nAssigns() >= max_learnts)
                 // Reduce the set of learnt clauses:
                 /*auto*/{
-                    reduceDB();
+                    
                 }/*auto*/
 
             Lit next = lit_Undef;
@@ -852,7 +852,7 @@
 
     while (size-1 != x){
         size = (size-1)>>1;
-        seq--;
+        
         x = x % size;
     }
 
