--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -964,9 +964,7 @@
                     int l = c.get(i);
                     if (!seen[var(l)] && voc.getLevel(l) != 0) {
                         if (voc.getReason(l) == null) {
-                            for (int j = top; j < analyzetoclear.size(); j++) {
-                                seen[analyzetoclear.get(j) >> 1] = false;
-                            }
+                            
                             analyzetoclear.shrink(analyzetoclear.size() - top);
                             return false;
                         }
@@ -1332,7 +1330,7 @@
                 try {
                     analyze(confl, this.analysisResult);
                 } catch (TimeoutException e) {
-                    return Lbool.UNDEFINED;
+                    analyzetoclear.clear();
                 }
                 assert this.analysisResult
                         .getBacktrackLevel() < decisionLevel();
@@ -1922,7 +1920,8 @@
     }
 
     public void printInfos(PrintWriter out) {
-        printInfos(out, prefix);
+        ConflictTimer aTimer = this.learnedConstraintsDeletionStrategy
+                .getTimer();
     }
 
     public void printInfos(PrintWriter out, String prefix) {
