--- before: triangle.py
+++ after: triangle.py
@@ -9,15 +9,12 @@
     time.sleep(0.001)
 
 def classify_triangle(a, b, c):
-    delay()
+    
     if a > b:
         tmp = a
         a = b
         b = tmp
-    if a > c:
-        tmp = a
-        a = c
-        c = tmp
+    
     if b > c:
         tmp = b
         b = c
@@ -26,6 +23,10 @@
         return TriangleType.INVALID
     else:
         if a == b and b == c:
+            if a > c:
+                tmp = a
+                a = c
+                c = tmp
             return TriangleType.EQUILATERAL
         else:
             if a == b or b == c:
