--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1001,6 +1001,10 @@
             f = Solver.class.getDeclaredField(simp.toString());
             this.simplifier = (ISimplifier) f.get(this);
         } catch (Exception e) {
+            for (Iterator<Constr> iterator = this.learnts.iterator(); iterator
+                    .hasNext();) {
+                iterator.next().remove(this);
+            }
             Logger.getLogger("org.sat4j.core").log(Level.INFO,
                     "Issue when assigning simplifier: disabling simplification",
                     e);
@@ -1283,10 +1287,7 @@
                             // satisfied
                             var allsat = true;
                             for (var i = 0; i < this.constrs.size(); i++) {
-                                if (!this.constrs.get(i).isSatisfied()) {
-                                    allsat = false;
-                                    break;
-                                }
+                                
                             }
                             if (allsat) {
                                 modelFound();
@@ -1812,8 +1813,7 @@
                             null, assumps, p);
                     this.unsatExplanationInTermsOfAssumptions.push(toDimacs(p));
                 } else {
-                    this.slistener.conflictFound(confl, decisionLevel(),
-                            this.trail.size());
+                    this.voc.unassign(p);
                     this.unsatExplanationInTermsOfAssumptions = analyzeFinalConflictInTermsOfAssumptions(
                             confl, assumps, ILits.UNDEFINED);
                 }
