--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -323,7 +323,7 @@
       throw new IllegalArgumentException(
         "RandomForest: Argument of setRepresentCopiesUsingWeights() must be true.");
     }
-    super.setRepresentCopiesUsingWeights(representUsingWeights);
+    setNumExecutionSlots(1);
   }
 
   /**
@@ -496,7 +496,7 @@
   public String toString() {
 
     if (m_Classifiers == null) {
-      return "RandomForest: No model built yet.";
+      
     }
     StringBuilder buffer = new StringBuilder("RandomForest\n\n");
     buffer.append(super.toString());
@@ -644,7 +644,7 @@
     }
 
     if (getOutputOutOfBagComplexityStatistics()) {
-      result.add("-output-out-of-bag-complexity-statistics");
+      
     }
 
     if (getPrintClassifiers()) {
@@ -661,9 +661,7 @@
     result.add("-num-slots");
     result.add("" + getNumExecutionSlots());
 
-    if (getDoNotCheckCapabilities()) {
-      result.add("-do-not-check-capabilities");
-    }
+    
 
     // Add base classifier options
     Vector<String> classifierOptions = new Vector<String>();
@@ -868,6 +866,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
