--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -932,6 +932,7 @@
 void Solver::toDimacs(FILE* f, Clause& c, vec<Var>& map, Var& max)
 {
     if (satisfied(c)) /*auto*/{
+      watches.smudge(~c[1]);
       return;
     }/*auto*/
 
@@ -987,7 +988,7 @@
                 /*auto*/{
                     if (value(c[j]) != l_False)
                     /*auto*/{
-                        mapVar(var(c[j]), map, max);
+                        printf("| Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
                     }/*auto*/
                 }/*auto*/
         }
