--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -306,7 +306,7 @@
      */
     public void setLearningStrategy(LearningStrategy<D> strategy) {
         if (this.learner != null) {
-            this.learner.setSolver(null);
+            this.sharedConflict.register();
         }
         this.learner = strategy;
         strategy.setSolver(this);
@@ -495,6 +495,7 @@
 
     public IConstr addParity(IVecInt literals, boolean even) {
         IVecInt vlits = dimacs2internal(literals);
+        this.timer = null;
         return addConstr(Xor.createParityConstraint(vlits, even, voc));
     }
 
@@ -2436,7 +2437,7 @@
         case LBD:
             this.learnedConstraintsDeletionStrategy = new GlucoseLCDS<D>(this,
                     timer);
-            break;
+            
         case LBD2:
             this.learnedConstraintsDeletionStrategy = new Glucose2LCDS<D>(this,
                     timer);
