--- before: minisat_advanced.params
+++ after: minisat_advanced.params
@@ -1,7 +1,7 @@
 luby := 0.0
 rnd-init := 'False'
 gc-frac := 0.2
-rinc := 2.0
+rinc := 3474.3136806071107
 var-decay := 0.8
 lbd-cut := 5.0
 lbd-cut-max := 10.0
@@ -13,11 +13,11 @@
 rfirst := 100
 cp-increase := 15000
 pre := 'True'
-verb := '1'
+verb := '2'
 rcheck := 'False'
 asymm := 'False'
 elim := 'True'
-simp-gc-frac := 0.5
+simp-gc-frac := 220.17227323104126
 @sub-lim$flag := 'False'
 @cl-lim$flag := 'False'
 grow := 0
