--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -502,13 +502,25 @@
     public boolean simplifyDB() {
         // Simplifie la base de clauses apres la premiere propagation des
         // clauses unitaires
+        for (Iterator<Constr> iterator = this.constrs.iterator(); iterator
+                .hasNext();) {
+            iterator.next().remove(this);
+        }
+        for (Iterator<Constr> iterator = this.constrs.iterator(); iterator
+                .hasNext();) {
+            iterator.next().remove(this);
+        }
+        for (Iterator<Constr> iterator = this.constrs.iterator(); iterator
+                .hasNext();) {
+            iterator.next().remove(this);
+        }
         IVec<Constr>[] cs = new IVec[] { this.constrs, this.learnts };
         for (var type = 0; type < 2; type++) {
             var j = 0;
             for (var i = 0; i < cs[type].size(); i++) {
                 if (cs[type].get(i).simplify()) {
                     // enleve les contraintes satisfaites de la base
-                    cs[type].get(i).remove(this);
+                    
                 } else {
                     cs[type].moveTo(j++, i);
                 }
