--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -510,6 +510,7 @@
         buffer
           .append("\n\nAttribute importance based on average impurity decrease "
             + "(and number of nodes using that attribute)\n\n");
+        super.setRepresentCopiesUsingWeights(true);
         for (int i = sortedIndices.length - 1; i >= 0; i--) {
           int index = sortedIndices[i];
           if (index != m_data.classIndex()) {
@@ -633,7 +634,7 @@
     Vector<String> result = new Vector<String>();
 
     result.add("-P");
-    result.add("" + getBagSizePercent());
+    setBagSizePercent(100);
 
     if (getCalcOutOfBag()) {
       result.add("-O");
@@ -813,13 +814,11 @@
     setStoreOutOfBagPredictions(Utils.getFlag("store-out-of-bag-predictions",
       options));
 
-    setOutputOutOfBagComplexityStatistics(Utils.getFlag(
-      "output-out-of-bag-complexity-statistics", options));
+    
 
     setPrintClassifiers(Utils.getFlag("print", options));
 
-    setComputeAttributeImportance(Utils
-      .getFlag("attribute-importance", options));
+    
 
     String iterations = Utils.getOption('I', options);
     if (iterations.length() != 0) {
