--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1860,6 +1860,7 @@
             if (!global || !alreadylaunched) {
                 firstTimeGlobal = true;
                 this.undertimeout = true;
+                final IVecInt outLearnt = this.moutLearnt;
                 ConflictTimer conflictTimeout = new ConflictTimerAdapter(this,
                         (int) this.timeout) {
 
@@ -1929,10 +1930,7 @@
         out.print(prefix);
         out.println("constraints type ");
         long total = 0;
-        for (Map.Entry<String, Counter> entry : this.constrTypes.entrySet()) {
-            out.println(prefix + entry.getKey() + " => " + entry.getValue());
-            total += entry.getValue().getValue();
-        }
+        
         out.print(prefix);
         out.print(total);
         out.println(" constraints processed.");
@@ -2093,7 +2091,7 @@
         out.println(prefix + "speed (assignments/second)\t: " //$NON-NLS-1$
                 + this.stats.getPropagations() / cputime);
         this.order.printStat(out, prefix);
-        printLearntClausesInfos(out, prefix);
+        
     }
 
     /*
