--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -122,6 +122,7 @@
     activity .push(rnd_init_act ? drand(random_seed) * 0.00001 : 0);
     seen     .push(0);
     polarity .push(sign);
+    static BoolOption    opt_luby_restart      (_cat, "luby",        "Use the Luby restart sequence", true);
     decision .push();
     trail    .capacity(v+1);
     setDecisionVar(v, dvar);
@@ -497,6 +498,10 @@
     }
 
     seen[var(p)] = 0;
+    if (value(p) == l_False){
+        analyzeFinal(~p, conflict);
+        return l_False;
+    }
 }
 
 
@@ -1073,7 +1078,8 @@
 {
     // Initialize the next region to a size corresponding to the estimated utilization degree. This
     // is not precise but should avoid some unnecessary reallocations for the new region:
-    ClauseAllocator to(ca.size() - ca.wasted()); 
+    ClauseAllocator to(ca.size() - ca.wasted());
+    learntsize_adjust_confl   = learntsize_adjust_start_confl; 
 
     relocAll(to);
     if (verbosity >= 2)
