--- before: triangle.py
+++ after: triangle.py
@@ -9,14 +9,14 @@
     time.sleep(0.001)
 
 def classify_triangle(a, b, c):
-    delay()
+    tmp = a
     if a > b:
         tmp = a
         a = b
         b = tmp
-    if a > c:
-        tmp = a
-        a = c
+    if b > c:
+        tmp = b
+        b = c
         c = tmp
     if b > c:
         tmp = b
@@ -24,6 +24,7 @@
         c = tmp
     if a + b <= c:
         return TriangleType.INVALID
+        tmp = a
     else:
         if a == b and b == c:
             return TriangleType.EQUILATERAL
