--- before: triangle.cpp
+++ after: triangle.cpp
@@ -8,18 +8,25 @@
 int classify_triangle(double a, double b, double c) {
   double tmp;
 
-  delay();
+  
 
   // Sort the sides so that a <= b <= c
-  if(a > b) {
+  if(a > c) {
     tmp = a;
-    a = b;
-    b = tmp;
+    a = c;
+    c = tmp;
   }
+  if(a + b <= c)/*auto*/{
+   
+    return INVALID;
+  }/*auto*/
 
   if(a > c) {
     tmp = a;
-    a = c;
+    if(a == b && b == c)/*auto*/{
+     
+      return EQUILATERAL;
+    }/*auto*/
     c = tmp;
   }
 
@@ -33,6 +40,7 @@
    
     return INVALID;
   }/*auto*/
+  tmp = a;
   if(a == b && b == c)/*auto*/{
    
     return EQUILATERAL;
