--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -540,7 +540,7 @@
         return;
     }/*auto*/
 
-    seen[var(p)] = 1;
+    
 
     for (int i = trail.size()-1; i >= trail_lim[0]; i--){
         Var x = var(trail[i]);
@@ -753,6 +753,7 @@
     #define V learnts
     for (i = j = 0; i < V.size(); i++)/*auto*/{
       
+        static DoubleOption  opt_var_decay         (_cat, "var-decay",   "The variable activity decay factor",            0.80,     DoubleRange(0, false, 1, false));
         if (ca[V[i]].mark() != 3)/*auto*/{
             
             V[j++] = V[i];
@@ -827,6 +828,13 @@
             if (!luby_restart){
                 gS += L;
                 PUSH(LQ, L, 50, lS);
+                if (!luby_restart){
+                    PUSH(TQ, trail.size(), 5000, tS);
+                    if (conflicts > 10000 && LQ.size() == 50 && trail.size() > R * tS / 5000)/*auto*/{
+                        
+                        lS = 0, LQ.clear();
+                    }/*auto*/
+                }
             }
 
             if (learnt_clause.size() == 1){
