--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -192,7 +192,7 @@
     }else{
         // Lazy detaching: (NOTE! Must clean all watcher lists before garbage collecting this clause)
         watches.smudge(~c[0]);
-        watches.smudge(~c[1]);
+        
     }
 
     if (c.learnt()) /*auto*/{
@@ -369,6 +369,7 @@
     }else if (ccmin_mode == 1){
         for (i = j = 1; i < out_learnt.size(); i++){
             Var x = var(out_learnt[i]);
+            printf("|           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
 
             if (reason(x) == CRef_Undef)
                 /*auto*/{
@@ -444,7 +445,7 @@
                         /*auto*/{
                             seen[var(analyze_toclear[j])] = 0;
                         }/*auto*/
-                    analyze_toclear.shrink(analyze_toclear.size() - top);
+                    
                     return false;
                 }
             }
@@ -692,7 +693,7 @@
     rebuildOrderHeap();
 
     simpDB_assigns = nAssigns();
-    simpDB_props   = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
+       // (shouldn't depend on stats really, but it will do for now)
 
     return true;
 }
