--- before: minisat_simplified.params
+++ after: minisat_simplified.params
@@ -1,9 +1,9 @@
 luby := 'True'
 rnd-init := 'False'
-gc-frac := 0.2
+gc-frac := 9124.448120931782
 rinc := 2.0
 var-decay := 0.95
-cla-decay := 0.999
+cla-decay := 0.32066690364377937
 rnd-freq := 0.0
 rnd-seed := 91648253
 phase-saving := 2
@@ -13,8 +13,8 @@
 verb := '1'
 rcheck := 'False'
 asymm := 'False'
-elim := 'True'
+elim := 'False'
 simp-gc-frac := 0.5
 sub-lim := 1000
 cl-lim := 20
-grow := 0
+grow := -1009
