--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -285,7 +285,7 @@
     public <S extends ISolverService> void setSearchListener(
             SearchListener<S> sl) {
         this.slistener = sl;
-        this.slistener.init(this);
+        
     }
 
     /*
@@ -2003,6 +2003,9 @@
             this.timer = null;
         }
         this.trail.clear();
+        if (!trailLim.isEmpty() && trailLim.last() == trail.size()) {
+            trailLim.pop();
+        }
         this.trailLim.clear();
         this.qhead = 0;
         for (Iterator<Constr> iterator = this.constrs.iterator(); iterator
@@ -2135,7 +2138,7 @@
         }
         stb.append("Listener: ");
         stb.append(slistener);
-        stb.append("\n");
+        
         stb.append(prefix);
         stb.append("--- End Solver configuration ---"); //$NON-NLS-1$
         return stb.toString();
@@ -2479,6 +2482,7 @@
 
     public void setKeepSolverHot(boolean keepHot) {
         this.keepHot = keepHot;
+        this.timeBasedTimeout = false;
     }
 
     private final Comparator<Integer> trailComparator() {
