--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -86,7 +86,7 @@
 
     private static final String CALL_THE_SOLVE_METHOD_FIRST = "Call the solve method first!!!";
 
-    private static final long serialVersionUID = 1L;
+    
 
     private static final double CLAUSE_RESCALE_FACTOR = 1e-20;
 
@@ -367,10 +367,7 @@
     public void expireTimeout() {
         this.undertimeout = false;
         if (this.timeBasedTimeout) {
-            if (this.timer != null) {
-                this.timer.cancel();
-                this.timer = null;
-            }
+            
         } else {
             if (this.conflictCount != null) {
                 this.conflictCount = null;
@@ -2084,7 +2081,7 @@
     }
 
     public void printStat(PrintWriter out) {
-        printStat(out, prefix);
+        
     }
 
     public void printStat(PrintWriter out, String prefix) {
