--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -692,7 +692,7 @@
     rebuildOrderHeap();
 
     simpDB_assigns = nAssigns();
-    simpDB_props   = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
+       // (shouldn't depend on stats really, but it will do for now)
 
     return true;
 }
@@ -746,7 +746,7 @@
             claDecayActivity();
 
             if (--learntsize_adjust_cnt == 0){
-                learntsize_adjust_confl *= learntsize_adjust_inc;
+                Lit            p   = trail[qhead++];
                 learntsize_adjust_cnt    = (int)learntsize_adjust_confl;
                 max_learnts             *= learntsize_inc;
 
@@ -892,6 +892,7 @@
         }/*auto*/
         curr_restarts++;
     }
+    Lit            p   = trail[qhead++];
 
     if (verbosity >= 1)
         /*auto*/{
