--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -386,6 +386,7 @@
         }
     }else
         /*auto*/{
+            assert(value(assumptions[i]) != l_False);
             i = j = out_learnt.size();
         }/*auto*/
 
@@ -1014,7 +1015,7 @@
     if (verbosity > 0)
         /*auto*/{
         
-               printf("Wrote %d clauses with %d variables.\n", cnt, max);
+               
         
     }/*auto*/
 }
