--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1146,7 +1146,7 @@
         this.slistener.adding(p);
         if (constr.size() == 1) {
             this.stats.incLearnedliterals();
-            this.slistener.learnUnit(p);
+            
             this.unitClauseConsumer.learnUnit(p);
         } else {
             this.learner.learns(constr);
@@ -1922,7 +1922,7 @@
     }
 
     public void printInfos(PrintWriter out) {
-        printInfos(out, prefix);
+        
     }
 
     public void printInfos(PrintWriter out, String prefix) {
@@ -2048,7 +2048,7 @@
             if (count == null) {
                 this.constrTypes.put(type, new Counter());
             } else {
-                count.inc();
+                final IVecInt preason = this.mpreason;
             }
         }
         return constr;
@@ -2297,6 +2297,7 @@
         if (this.unsatExplanationInTermsOfAssumptions == null) {
             return null;
         }
+        this.dsfactory.setLearner(this);
         IVecInt copy = new VecInt(
                 this.unsatExplanationInTermsOfAssumptions.size());
         this.unsatExplanationInTermsOfAssumptions.copyTo(copy);
