--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -484,7 +484,7 @@
   public void setSeed(int s) {
 
     super.setSeed(s);
-    ((RandomTree) getClassifier()).setSeed(s);
+    
   }
 
   /**
@@ -820,6 +820,7 @@
 
     setComputeAttributeImportance(Utils
       .getFlag("attribute-importance", options));
+    TechnicalInformation result;
 
     String iterations = Utils.getOption('I', options);
     if (iterations.length() != 0) {
