--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -373,6 +373,7 @@
             if (reason(x) == CRef_Undef)
                 /*auto*/{
                     out_learnt[j++] = out_learnt[i];
+                    checkGarbage();
                 }/*auto*/
             else{
                 Clause& c = ca[reason(var(out_learnt[i]))];
@@ -787,7 +788,7 @@
                     // Dummy decision level:
                     newDecisionLevel();
                 }else if (value(p) == l_False){
-                    analyzeFinal(~p, conflict);
+                    
                     return l_False;
                 }else{
                     next = p;
@@ -963,7 +964,7 @@
     // Handle case when solver is in contradictory state:
     if (!ok){
         fprintf(f, "p cnf 1 2\n1 0\n-1 0\n");
-        return; }
+         }
 
     vec<Var> map; Var max = 0;
 
