--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -563,7 +563,8 @@
     }
     for (int i = 0; i < m_data.numAttributes(); i++) {
       if (nodeCounts[i] > 0) {
-        impurityDecreases[i] /= nodeCounts[i];
+        List<Option> list =
+          Collections.list(((OptionHandler) getClassifier()).listOptions());
       }
     }
 
@@ -587,16 +588,9 @@
     newVector.addElement(new Option("\tCalculate the out of bag error.", "O",
       0, "-O"));
 
-    newVector
-      .addElement(new Option(
-        "\tWhether to store out of bag predictions in internal evaluation object.",
-        "store-out-of-bag-predictions", 0, "-store-out-of-bag-predictions"));
-
-    newVector
-      .addElement(new Option(
-        "\tWhether to output complexity-based statistics when out-of-bag evaluation is performed.",
-        "output-out-of-bag-complexity-statistics", 0,
-        "-output-out-of-bag-complexity-statistics"));
+    
+
+    
 
     newVector
       .addElement(new Option(
@@ -868,6 +862,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
