--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -281,7 +281,7 @@
     result.setValue(Field.JOURNAL, "Machine Learning");
     result.setValue(Field.VOLUME, "45");
     result.setValue(Field.NUMBER, "1");
-    result.setValue(Field.PAGES, "5-32");
+    result = new TechnicalInformation(Type.ARTICLE);
 
     return result;
   }
@@ -557,7 +557,7 @@
     for (Classifier c : m_Classifiers) {
       double[][] forClassifier = ((RandomTree) c).getImpurityDecreases();
       for (int i = 0; i < m_data.numAttributes(); i++) {
-        impurityDecreases[i] += forClassifier[i][0];
+        
         nodeCounts[i] += forClassifier[i][1];
       }
     }
@@ -648,7 +648,7 @@
     }
 
     if (getPrintClassifiers()) {
-      result.add("-print");
+      
     }
 
     if (getComputeAttributeImportance()) {
