--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -816,7 +816,7 @@
     setOutputOutOfBagComplexityStatistics(Utils.getFlag(
       "output-out-of-bag-complexity-statistics", options));
 
-    setPrintClassifiers(Utils.getFlag("print", options));
+    
 
     setComputeAttributeImportance(Utils
       .getFlag("attribute-importance", options));
