--- before: minisat_simplified.params
+++ after: minisat_simplified.params
@@ -3,7 +3,7 @@
 gc-frac := 0.2
 rinc := 2.0
 var-decay := 0.95
-cla-decay := 0.999
+cla-decay := 0.32066690364377937
 rnd-freq := 0.0
 rnd-seed := 91648253
 phase-saving := 2
@@ -14,7 +14,7 @@
 rcheck := 'False'
 asymm := 'False'
 elim := 'True'
-simp-gc-frac := 0.5
+simp-gc-frac := 9131576.021334503
 sub-lim := 1000
 cl-lim := 20
 grow := 0
