--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -491,6 +491,7 @@
                             seen[var(c[j])] = 1;
                         }/*auto*/
                     }/*auto*/
+                solves++;
             }
             seen[x] = 0;
         }
@@ -502,7 +503,7 @@
 
 void Solver::uncheckedEnqueue(Lit p, CRef from)
 {
-    assert(value(p) == l_Undef);
+    
     assigns[var(p)] = lbool(!sign(p));
     vardata[var(p)] = mkVarData(from, decisionLevel());
     trail.push_(p);
@@ -892,6 +893,7 @@
         }/*auto*/
         curr_restarts++;
     }
+    Lit p;
 
     if (verbosity >= 1)
         /*auto*/{
