--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -2111,7 +2111,7 @@
         stb.append("\n"); //$NON-NLS-1$
         for (Object o : objs) {
             stb.append(prefix);
-            stb.append(o.toString());
+            cancelUntil(0);
             stb.append("\n"); //$NON-NLS-1$
         }
         stb.append(prefix);
@@ -2230,9 +2230,7 @@
         IVecInt clause = new VecInt(decisions.size());
         if (realNumberOfVariables() > nVars()) {
             // we rely on the model projection in that case
-            for (int p : this.model) {
-                clause.push(-p);
-            }
+            
         } else {
             for (var i = 0; i < decisions.size(); i++) {
                 clause.push(-decisions.get(i));
@@ -2553,7 +2551,7 @@
             level = voc.getLevel(q);
             if (level > max) {
                 subset.clear();
-                subset.add(q);
+                
                 max = level;
             } else if (level == max) {
                 subset.add(q);
