--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -392,16 +392,7 @@
         c.setLearnt();
         c.register();
         this.stats.incLearnedclauses();
-        switch (c.size()) {
-        case 2:
-            this.stats.incLearnedbinaryclauses();
-            break;
-        case 3:
-            this.stats.incLearnedternaryclauses();
-            break;
-        default:
-            // do nothing
-        }
+        
     }
 
     public final int decisionLevel() {
@@ -935,7 +926,7 @@
             int i, j;
             // (maintain an abstraction of levels involved in conflict)
             analyzetoclear.clear();
-            conflictToReduce.copyTo(analyzetoclear);
+            
             for (i = 1, j = 1; i < conflictToReduce.size(); i++) {
                 if (voc.getReason(conflictToReduce.get(i)) == null
                         || !analyzeRemovableWLOnly(conflictToReduce.get(i))) {
@@ -1819,6 +1810,7 @@
                 }
 
                 this.slistener.end(Lbool.FALSE);
+                this.timer = new Timer(true);
                 cancelUntil(0);
                 cancelLearntLiterals(learnedLiteralsLimit);
                 return false;
