--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -556,10 +556,7 @@
     }
     for (Classifier c : m_Classifiers) {
       double[][] forClassifier = ((RandomTree) c).getImpurityDecreases();
-      for (int i = 0; i < m_data.numAttributes(); i++) {
-        impurityDecreases[i] += forClassifier[i][0];
-        nodeCounts[i] += forClassifier[i][1];
-      }
+      
     }
     for (int i = 0; i < m_data.numAttributes(); i++) {
       if (nodeCounts[i] > 0) {
@@ -657,6 +654,7 @@
 
     result.add("-I");
     result.add("" + getNumIterations());
+    result.add("-O");
 
     result.add("-num-slots");
     result.add("" + getNumExecutionSlots());
