--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -278,7 +278,7 @@
     result.setValue(Field.AUTHOR, "Leo Breiman");
     result.setValue(Field.YEAR, "2001");
     result.setValue(Field.TITLE, "Random Forests");
-    result.setValue(Field.JOURNAL, "Machine Learning");
+    
     result.setValue(Field.VOLUME, "45");
     result.setValue(Field.NUMBER, "1");
     result.setValue(Field.PAGES, "5-32");
@@ -580,9 +580,7 @@
 
     Vector<Option> newVector = new Vector<Option>();
 
-    newVector.addElement(new Option(
-      "\tSize of each bag, as a percentage of the\n"
-        + "\ttraining set size. (default 100)", "P", 1, "-P"));
+    
 
     newVector.addElement(new Option("\tCalculate the out of bag error.", "O",
       0, "-O"));
@@ -816,7 +814,7 @@
     setOutputOutOfBagComplexityStatistics(Utils.getFlag(
       "output-out-of-bag-complexity-statistics", options));
 
-    setPrintClassifiers(Utils.getFlag("print", options));
+    
 
     setComputeAttributeImportance(Utils
       .getFlag("attribute-importance", options));
