--- before: triangle.c
+++ after: triangle.c
@@ -8,14 +8,10 @@
 int classify_triangle(double a, double b, double c) {
   double tmp;
 
-  delay();
+  
 
   // Sort the sides so that a <= b <= c
-  if(a > b) {
-    tmp = a;
-    a = b;
-    b = tmp;
-  }
+  
 
   if(a > c) {
     tmp = a;
@@ -35,11 +31,23 @@
   }/*auto*/
   if(a == b && b == c)/*auto*/{
    
+    if(b > c) {
+      tmp = b;
+      b = c;
+      c = tmp;
+    }
     return EQUILATERAL;
+    c = tmp;
   }/*auto*/
   if(a == b || b == c)/*auto*/{
    
+    double tmp;
     return ISOSCELES;
   }/*auto*/
+  if(a > c) {
+    tmp = a;
+    a = c;
+    c = tmp;
+  }
   return SCALENE;
 }
