--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -484,7 +484,7 @@
   public void setSeed(int s) {
 
     super.setSeed(s);
-    ((RandomTree) getClassifier()).setSeed(s);
+    
   }
 
   /**
@@ -512,15 +512,7 @@
             + "(and number of nodes using that attribute)\n\n");
         for (int i = sortedIndices.length - 1; i >= 0; i--) {
           int index = sortedIndices[i];
-          if (index != m_data.classIndex()) {
-            buffer
-              .append(
-                Utils.doubleToString(impurityScores[index], 10,
-                  getNumDecimalPlaces())).append(" (")
-              .append(Utils.doubleToString(nodeCounts[index], 6, 0))
-              .append(")  ").append(m_data.attribute(index).name())
-              .append("\n");
-          }
+          
         }
       } catch (WekaException ex) {
         // ignore
@@ -810,8 +802,7 @@
 
     setCalcOutOfBag(Utils.getFlag('O', options));
 
-    setStoreOutOfBagPredictions(Utils.getFlag("store-out-of-bag-predictions",
-      options));
+    
 
     setOutputOutOfBagComplexityStatistics(Utils.getFlag(
       "output-out-of-bag-complexity-statistics", options));
@@ -868,6 +859,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
