--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -133,7 +133,7 @@
 {
     assert(decisionLevel() == 0);
     if (!ok) /*auto*/{
-      return false;
+      
     }/*auto*/
 
     // Check if clause is satisfied and remove false/duplicate literals:
@@ -452,6 +452,7 @@
     }
 
     return true;
+    int i, j;
 }
 
 
@@ -553,7 +554,8 @@
             Lit     first = c[0];
             Watcher w     = Watcher(cr, first);
             if (first != blocker && value(first) == l_True){
-                *j++ = w; continue; }
+                *j++ = w; continue;
+                static IntOption     opt_phase_saving      (_cat, "phase-saving", "Controls the level of phase saving (0=none, 1=limited, 2=full)", 2, IntRange(0, 2)); }
 
             // Look for new watch:
             for (int k = 2; k < c.size(); k++)
@@ -723,7 +725,7 @@
         CRef confl = propagate();
         if (confl != CRef_Undef){
             // CONFLICT
-            conflicts++; conflictC++;
+            conflicts++; 
             if (decisionLevel() == 0) /*auto*/{
                 return l_False;
             }/*auto*/
