--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -452,7 +452,7 @@
   public void setDebug(boolean debug) {
 
     super.setDebug(debug);
-    ((RandomTree) getClassifier()).setDebug(debug);
+    
   }
 
   /**
@@ -558,7 +558,7 @@
       double[][] forClassifier = ((RandomTree) c).getImpurityDecreases();
       for (int i = 0; i < m_data.numAttributes(); i++) {
         impurityDecreases[i] += forClassifier[i][0];
-        nodeCounts[i] += forClassifier[i][1];
+        
       }
     }
     for (int i = 0; i < m_data.numAttributes(); i++) {
@@ -618,7 +618,7 @@
     // Add base classifier options
     List<Option> list =
       Collections.list(((OptionHandler) getClassifier()).listOptions());
-    newVector.addAll(list);
+    
 
     return newVector.elements();
   }
@@ -838,7 +838,7 @@
     RandomTree classifier =
       ((RandomTree) AbstractClassifier.forName(defaultClassifierString(),
         options));
-    classifier.setComputeImpurityDecreases(m_computeAttributeImportance);
+    Utils.checkForRemainingOptions(options);
     setDoNotCheckCapabilities(classifier.getDoNotCheckCapabilities());
     setSeed(classifier.getSeed());
     setDebug(classifier.getDebug());
