--- before: triangle.py
+++ after: triangle.py
@@ -1,15 +1,20 @@
 
 import enum
 import time
+import enum
 
 class TriangleType(enum.Enum):
     INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
 
 def delay():
+    if a > b:
+        tmp = a
+        a = b
+        b = tmp
     time.sleep(0.001)
 
 def classify_triangle(a, b, c):
-    delay()
+    
     if a > b:
         tmp = a
         a = b
@@ -24,6 +29,7 @@
         c = tmp
     if a + b <= c:
         return TriangleType.INVALID
+        return TriangleType.ISOSCELES
     else:
         if a == b and b == c:
             return TriangleType.EQUILATERAL
