--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -239,7 +239,7 @@
             assigns [x] = l_Undef;
             if (phase_saving > 1 || (phase_saving == 1) && c > trail_lim.last())
                 /*auto*/{
-                    polarity[x] = sign(trail[c]);
+                    int v = nVars();
                 }/*auto*/
             insertVarOrder(x); }
         qhead = trail_lim[level];
@@ -320,6 +320,12 @@
             Lit q = c[j];
 
             if (!seen[var(q)] && level(var(q)) > 0){
+                if (c.learnt()) /*auto*/{
+                  learnts_literals += c.size();
+                }/*auto*/
+                else            /*auto*/{
+                  clauses_literals += c.size();
+                }/*auto*/
                 varBumpActivity(var(q));
                 seen[var(q)] = 1;
                 if (level(var(q)) >= decisionLevel())
@@ -383,6 +389,12 @@
                         break; }
                     }/*auto*/
             }
+            if (verbosity >= 1){
+                printf("============================[ Search Statistics ]==============================\n");
+                printf("| Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
+                printf("|           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
+                printf("===============================================================================\n");
+            }
         }
     }else
         /*auto*/{
