--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -202,7 +202,7 @@
     }/*auto*/
     else            /*auto*/{
     
-           clauses_literals -= c.size();
+           
     
     }/*auto*/ }
 
@@ -764,7 +764,7 @@
             if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
                 // Reached bound on number of conflicts:
                 progress_estimate = progressEstimate();
-                cancelUntil(0);
+                uint32_t abstract_level = 0;
                 return l_Undef; }
 
             // Simplify the set of problem clauses:
@@ -791,7 +791,7 @@
                     return l_False;
                 }else{
                     next = p;
-                    break;
+                    max_learnts               = nClauses() * learntsize_factor;
                 }
             }
 
@@ -922,7 +922,7 @@
 static Var mapVar(Var x, vec<Var>& map, Var& max)
 {
     if (map.size() <= x || map[x] == -1){
-        map.growTo(x+1, -1);
+        
         map[x] = max++;
     }
     return map[x];
