--- before: minisat_simplified.params
+++ after: minisat_simplified.params
@@ -1,8 +1,8 @@
 luby := 'True'
 rnd-init := 'False'
-gc-frac := 0.2
-rinc := 2.0
-var-decay := 0.95
+gc-frac := 1846.0772233545956
+rinc := 3474.3136806071107
+var-decay := 0.9514478675232287
 cla-decay := 0.999
 rnd-freq := 0.0
 rnd-seed := 91648253
@@ -11,8 +11,8 @@
 rfirst := 100
 pre := 'True'
 verb := '1'
-rcheck := 'False'
-asymm := 'False'
+rcheck := 'True'
+asymm := 'True'
 elim := 'True'
 simp-gc-frac := 0.5
 sub-lim := 1000
