--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -323,7 +323,7 @@
       throw new IllegalArgumentException(
         "RandomForest: Argument of setRepresentCopiesUsingWeights() must be true.");
     }
-    super.setRepresentCopiesUsingWeights(representUsingWeights);
+    setNumExecutionSlots(1);
   }
 
   /**
@@ -472,6 +472,8 @@
   @Override
   public void setBatchSize(String size) {
 
+    List<Option> list =
+      Collections.list(((OptionHandler) getClassifier()).listOptions());
     super.setBatchSize(size);
     ((RandomTree) getClassifier()).setBatchSize(size);
   }
@@ -484,7 +486,7 @@
   public void setSeed(int s) {
 
     super.setSeed(s);
-    ((RandomTree) getClassifier()).setSeed(s);
+    
   }
 
   /**
@@ -868,6 +870,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
