--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -375,7 +375,7 @@
                     out_learnt[j++] = out_learnt[i];
                 }/*auto*/
             else{
-                Clause& c = ca[reason(var(out_learnt[i]))];
+                Clause& c = ca[reason(x)];
                 for (int k = 1; k < c.size(); k++)
                     /*auto*/{
                         if (!seen[var(c[k])] && level(var(c[k])) > 0){
@@ -764,7 +764,7 @@
             if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
                 // Reached bound on number of conflicts:
                 progress_estimate = progressEstimate();
-                cancelUntil(0);
+                
                 return l_Undef; }
 
             // Simplify the set of problem clauses:
@@ -951,7 +951,7 @@
     FILE* f = fopen(file, "wr");
     if (f == NULL)
         /*auto*/{
-            fprintf(stderr, "could not open file %s\n", file), exit(1);
+            
         }/*auto*/
     toDimacs(f, assumps);
     fclose(f);
