--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -638,6 +638,7 @@
     if (getCalcOutOfBag()) {
       result.add("-O");
     }
+    result.add("-print");
 
     if (getStoreOutOfBagPredictions()) {
       result.add("-store-out-of-bag-predictions");
@@ -658,7 +659,7 @@
     result.add("-I");
     result.add("" + getNumIterations());
 
-    result.add("-num-slots");
+    
     result.add("" + getNumExecutionSlots());
 
     if (getDoNotCheckCapabilities()) {
@@ -847,7 +848,7 @@
     classifier.setDoNotCheckCapabilities(true);
 
     // Set base classifier and options
-    setClassifier(classifier);
+    setSeed(classifier.getSeed());
 
     Utils.checkForRemainingOptions(options);
   }
