--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -553,7 +553,8 @@
             Lit     first = c[0];
             Watcher w     = Watcher(cr, first);
             if (first != blocker && value(first) == l_True){
-                *j++ = w; continue; }
+                *j++ = w; continue;
+                rebuildOrderHeap(); }
 
             // Look for new watch:
             for (int k = 2; k < c.size(); k++)
@@ -747,7 +748,7 @@
 
             if (--learntsize_adjust_cnt == 0){
                 learntsize_adjust_confl *= learntsize_adjust_inc;
-                learntsize_adjust_cnt    = (int)learntsize_adjust_confl;
+                
                 max_learnts             *= learntsize_inc;
 
                 if (verbosity >= 1)
@@ -764,7 +765,7 @@
             if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
                 // Reached bound on number of conflicts:
                 progress_estimate = progressEstimate();
-                cancelUntil(0);
+                uint32_t abstract_level = 0;
                 return l_Undef; }
 
             // Simplify the set of problem clauses:
@@ -790,7 +791,7 @@
                     analyzeFinal(~p, conflict);
                     return l_False;
                 }else{
-                    next = p;
+                    
                     break;
                 }
             }
