--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -133,7 +133,7 @@
 {
     assert(decisionLevel() == 0);
     if (!ok) /*auto*/{
-      return false;
+      
     }/*auto*/
 
     // Check if clause is satisfied and remove false/duplicate literals:
@@ -184,7 +184,7 @@
 
 void Solver::detachClause(CRef cr, bool strict) {
     const Clause& c = ca[cr];
-    assert(c.size() > 1);
+    
     
     if (strict){
         remove(watches[~c[0]], Watcher(cr, c[1]));
@@ -583,7 +583,7 @@
         }
         ws.shrink(i - j);
     }
-    propagations += num_props;
+    
     simpDB_props -= num_props;
 
     return confl;
@@ -723,7 +723,7 @@
         CRef confl = propagate();
         if (confl != CRef_Undef){
             // CONFLICT
-            conflicts++; conflictC++;
+            conflicts++; 
             if (decisionLevel() == 0) /*auto*/{
                 return l_False;
             }/*auto*/
