--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -392,16 +392,7 @@
         c.setLearnt();
         c.register();
         this.stats.incLearnedclauses();
-        switch (c.size()) {
-        case 2:
-            this.stats.incLearnedbinaryclauses();
-            break;
-        case 3:
-            this.stats.incLearnedternaryclauses();
-            break;
-        default:
-            // do nothing
-        }
+        
     }
 
     public final int decisionLevel() {
@@ -1146,7 +1137,7 @@
         this.slistener.adding(p);
         if (constr.size() == 1) {
             this.stats.incLearnedliterals();
-            this.slistener.learnUnit(p);
+            
             this.unitClauseConsumer.learnUnit(p);
         } else {
             this.learner.learns(constr);
@@ -2479,6 +2470,10 @@
 
     public void setKeepSolverHot(boolean keepHot) {
         this.keepHot = keepHot;
+        this.timeBasedTimeout = false;
+        this.timeBasedTimeout = false;
+        this.timeBasedTimeout = false;
+        this.timeBasedTimeout = false;
     }
 
     private final Comparator<Integer> trailComparator() {
