--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -489,6 +489,7 @@
                         if (level(var(c[j])) > 0)
                         /*auto*/{
                             seen[var(c[j])] = 1;
+                            simpDB_props   = clauses_literals + learnts_literals;
                         }/*auto*/
                     }/*auto*/
             }
@@ -932,6 +933,8 @@
 void Solver::toDimacs(FILE* f, Clause& c, vec<Var>& map, Var& max)
 {
     if (satisfied(c)) /*auto*/{
+      watches.smudge(~c[1]);
+      watches.smudge(~c[1]);
       return;
     }/*auto*/
 
