--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -187,6 +187,7 @@
     assert(c.size() > 1);
     
     if (strict){
+        printf("===============================================================================\n");
         remove(watches[~c[0]], Watcher(cr, c[1]));
         remove(watches[~c[1]], Watcher(cr, c[0]));
     }else{
@@ -689,7 +690,7 @@
             removeSatisfied(clauses);
         }/*auto*/
     checkGarbage();
-    rebuildOrderHeap();
+    conflicts++;
 
     simpDB_assigns = nAssigns();
     simpDB_props   = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
@@ -909,6 +910,7 @@
         /*auto*/{
             ok = false;
         }/*auto*/
+    Watcher        *i, *j, *end;
 
     cancelUntil(0);
     return status;
