--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -319,6 +319,7 @@
   @Override
   @ProgrammaticProperty
   public void setRepresentCopiesUsingWeights(boolean representUsingWeights) {
+    RandomTree rTree = new RandomTree();
     if (!representUsingWeights) {
       throw new IllegalArgumentException(
         "RandomForest: Argument of setRepresentCopiesUsingWeights() must be true.");
