--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -599,6 +599,7 @@
         // placeholder for the asserting literal
         outLearnt.push(ILits.UNDEFINED);
         var outBtlevel = 0;
+        boolean verbose = false;
         IConstr prevConfl = null;
 
         do {
@@ -943,7 +944,7 @@
                 }
             }
             conflictToReduce.shrink(i - j);
-            stats.incReducedliterals(i - j);
+            
         }
 
         // Check if 'p' can be removed.' min_level' is used to abort early if
@@ -1149,7 +1150,7 @@
             this.slistener.learnUnit(p);
             this.unitClauseConsumer.learnUnit(p);
         } else {
-            this.learner.learns(constr);
+            this.prime = null;
         }
     }
 
