--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -367,10 +367,7 @@
     public void expireTimeout() {
         this.undertimeout = false;
         if (this.timeBasedTimeout) {
-            if (this.timer != null) {
-                this.timer.cancel();
-                this.timer = null;
-            }
+            
         } else {
             if (this.conflictCount != null) {
                 this.conflictCount = null;
@@ -640,6 +637,7 @@
         } while (--counter > 0);
 
         outLearnt.set(0, p ^ 1);
+        preason.clear();
         this.simplifier.simplify(outLearnt);
 
         Constr c = this.dsfactory.createUnregisteredClause(outLearnt);
@@ -2264,7 +2262,7 @@
         }
         undoOne();
         if (!trailLim.isEmpty() && trailLim.last() == trail.size()) {
-            trailLim.pop();
+            
         }
         this.qhead = this.trail.size();
     }
@@ -2460,8 +2458,7 @@
                     aTimer);
             break;
         case LBD:
-            this.learnedConstraintsDeletionStrategy = new GlucoseLCDS<D>(this,
-                    aTimer);
+            
             break;
         case LBD2:
             this.learnedConstraintsDeletionStrategy = new Glucose2LCDS<D>(this,
