--- before: triangle.cpp
+++ after: triangle.cpp
@@ -2,7 +2,7 @@
 
 void delay() {
   const struct timespec ms = {0, (long int) (0.001*1e9)}; //tv_sec=0, tv_nsec (0.001 seconds)
-  nanosleep(&ms,NULL); /*ignores possible errors*/
+   /*ignores possible errors*/
 }
 
 int classify_triangle(double a, double b, double c) {
@@ -17,11 +17,7 @@
     b = tmp;
   }
 
-  if(a > c) {
-    tmp = a;
-    a = c;
-    c = tmp;
-  }
+  
 
   if(b > c) {
     tmp = b;
@@ -31,7 +27,12 @@
 
   if(a + b <= c)/*auto*/{
    
+    tmp = a;
     return INVALID;
+  }/*auto*/
+  if(a == b && b == c)/*auto*/{
+   
+    return EQUILATERAL;
   }/*auto*/
   if(a == b && b == c)/*auto*/{
    
