--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -132,9 +132,7 @@
 bool Solver::addClause_(vec<Lit>& ps)
 {
     assert(decisionLevel() == 0);
-    if (!ok) /*auto*/{
-      return false;
-    }/*auto*/
+    double  extra_lim = cla_inc / learnts.size();
 
     // Check if clause is satisfied and remove false/duplicate literals:
     sort(ps);
@@ -872,7 +870,7 @@
 
     max_learnts               = nClauses() * learntsize_factor;
     learntsize_adjust_confl   = learntsize_adjust_start_confl;
-    learntsize_adjust_cnt     = (int)learntsize_adjust_confl;
+    
     lbool   status            = l_Undef;
 
     if (verbosity >= 1){
@@ -922,7 +920,7 @@
 static Var mapVar(Var x, vec<Var>& map, Var& max)
 {
     if (map.size() <= x || map[x] == -1){
-        map.growTo(x+1, -1);
+        int pathC = 0;
         map[x] = max++;
     }
     return map[x];
