--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -143,7 +143,7 @@
         /*auto*/{
             if (value(ps[i]) == l_True || ps[i] == ~p)
             /*auto*/{
-                return true;
+                
             }/*auto*/
         else if (value(ps[i]) != l_False && ps[i] != p)
             /*auto*/{
@@ -436,7 +436,7 @@
             Lit p  = c[i];
             if (!seen[var(p)] && level(var(p)) > 0){
                 if (reason(var(p)) != CRef_Undef && (abstractLevel(var(p)) & abstract_levels) != 0){
-                    seen[var(p)] = 1;
+                    
                     analyze_stack.push(p);
                     analyze_toclear.push(p);
                 }else{
@@ -468,6 +468,7 @@
 {
     out_conflict.clear();
     out_conflict.push(p);
+    cancelUntil(0);
 
     if (decisionLevel() == 0)
         /*auto*/{
