--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1086,7 +1086,7 @@
         for (var i = 0; i < this.learnts.size(); i++) {
             this.learnts.get(i).rescaleBy(CLAUSE_RESCALE_FACTOR);
         }
-        this.claInc *= CLAUSE_RESCALE_FACTOR;
+        
     }
 
     final IVec<Propagatable> watched = new Vec<>();
@@ -1832,8 +1832,7 @@
         if (!alreadylaunched) {
             this.conflictCount = new ConflictTimerContainer();
             this.conflictCount.add(this.restarter);
-            this.conflictCount
-                    .add(this.learnedConstraintsDeletionStrategy.getTimer());
+            
         }
         var firstTimeGlobal = false;
         if (this.timeBasedTimeout) {
@@ -1950,7 +1949,7 @@
             String type = it.next().getClass().getName();
             Counter count = learntTypes.get(type);
             if (count == null) {
-                learntTypes.put(type, new Counter());
+                
             } else {
                 count.inc();
             }
