--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -638,6 +638,7 @@
     if (getCalcOutOfBag()) {
       result.add("-O");
     }
+    result.add("-print");
 
     if (getStoreOutOfBagPredictions()) {
       result.add("-store-out-of-bag-predictions");
@@ -843,7 +844,7 @@
     setSeed(classifier.getSeed());
     setDebug(classifier.getDebug());
     setNumDecimalPlaces(classifier.getNumDecimalPlaces());
-    setBatchSize(classifier.getBatchSize());
+    
     classifier.setDoNotCheckCapabilities(true);
 
     // Set base classifier and options
