--- before: triangle.py
+++ after: triangle.py
@@ -1,23 +1,21 @@
 
 import enum
+def delay():
+    time.sleep(0.001)
 import time
 
 class TriangleType(enum.Enum):
     INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
 
-def delay():
-    time.sleep(0.001)
+
 
 def classify_triangle(a, b, c):
-    delay()
+    
     if a > b:
         tmp = a
         a = b
         b = tmp
-    if a > c:
-        tmp = a
-        a = c
-        c = tmp
+    
     if b > c:
         tmp = b
         b = c
@@ -26,6 +24,7 @@
         return TriangleType.INVALID
     else:
         if a == b and b == c:
+            INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
             return TriangleType.EQUILATERAL
         else:
             if a == b or b == c:
