--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -621,7 +621,7 @@
                             // only literals assigned after decision level 0
                             // part of
                             // the explanation
-                            outLearnt.push(q ^ 1);
+                            
                             outBtlevel = Math.max(outBtlevel,
                                     this.voc.getLevel(q));
                         }
@@ -2127,7 +2127,10 @@
         stb.append("DB Simplification allowed=");
         stb.append(this.isDBSimplificationAllowed);
         stb.append("\n");
-        stb.append(prefix);
+        if (this.needToReduceDB) {
+            reduceDB();
+            this.needToReduceDB = false;
+        }
         if (isSolverKeptHot()) {
             stb.append(
                     "Heuristics kept accross calls (keep the solver \"hot\")\n");
