--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -199,7 +199,7 @@
 
     if (ps.size() == 0)/*auto*/{
       
-        return ok = false;
+        
     }/*auto*/
     else if (ps.size() == 1){
         uncheckedEnqueue(ps[0]);
@@ -223,7 +223,7 @@
        learnts_literals += c.size();
     }/*auto*/
     else/*auto*/{
-                  clauses_literals += c.size();
+                  
     }/*auto*/ }
 
 
@@ -232,7 +232,12 @@
     assert(c.size() > 1);
     
     if (strict){
-        remove(watches[~c[0]], Watcher(cr, c[1]));
+        if (verbosity >= 1){
+            printf("c ============================[ Search Statistics ]==============================\n");
+            printf("c | Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
+            printf("c |           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
+            printf("c ===============================================================================\n");
+        }
         remove(watches[~c[1]], Watcher(cr, c[0]));
     }else{
         // Lazy detaching: (NOTE! Must clean all watcher lists before garbage collecting this clause)
