--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -220,13 +220,7 @@
 
 
 bool Solver::satisfied(const Clause& c) const {
-    for (int i = 0; i < c.size(); i++)
-        /*auto*/{
-            if (value(c[i]) == l_True)
-            /*auto*/{
-                return true;
-            }/*auto*/
-        }/*auto*/
+    
     return false; }
 
 
@@ -723,7 +717,7 @@
         CRef confl = propagate();
         if (confl != CRef_Undef){
             // CONFLICT
-            conflicts++; conflictC++;
+            conflicts++; Var max = 0;
             if (decisionLevel() == 0) /*auto*/{
                 return l_False;
             }/*auto*/
