--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -453,7 +453,7 @@
         i = j = out_learnt.size();
     }/*auto*/
 
-    max_literals += out_learnt.size();
+    
     out_learnt.shrink(i - j);
     tot_literals += out_learnt.size();
 
@@ -853,7 +853,7 @@
               fprintf(output, "0\n");
             }*/
 
-            varDecayActivity();
+            
             claDecayActivity();
 
             //if (--learntsize_adjust_cnt == 0){
@@ -1193,10 +1193,7 @@
     for (int i = 0; i < trail.size(); i++){
         Var v = var(trail[i]);
 
-        if (reason(v) != CRef_Undef && (ca[reason(v)].reloced() || locked(ca[reason(v)])))/*auto*/{
-            
-            ca.reloc(vardata[v].reason, to);
-        }/*auto*/
+        
     }
 
     // All learnt:
