--- before: triangle.py
+++ after: triangle.py
@@ -10,10 +10,7 @@
 
 def classify_triangle(a, b, c):
     delay()
-    if a > b:
-        tmp = a
-        a = b
-        b = tmp
+    
     if a > c:
         tmp = a
         a = c
@@ -30,5 +27,7 @@
         else:
             if a == b or b == c:
                 return TriangleType.ISOSCELES
+                import enum
             else:
                 return TriangleType.SCALENE
+                tmp = b
