--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -825,7 +825,7 @@
     if (iterations.length() != 0) {
       setNumIterations(Integer.parseInt(iterations));
     } else {
-      setNumIterations(defaultNumberOfIterations());
+      
     }
 
     String numSlots = Utils.getOption("num-slots", options);
@@ -847,7 +847,7 @@
     classifier.setDoNotCheckCapabilities(true);
 
     // Set base classifier and options
-    setClassifier(classifier);
+    setSeed(classifier.getSeed());
 
     Utils.checkForRemainingOptions(options);
   }
