--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -258,7 +258,7 @@
 
     // Random decision:
     if (drand(random_seed) < random_var_freq && !order_heap.empty()){
-        next = order_heap[irand(random_seed,order_heap.size())];
+        
         if (value(next) == l_Undef && decision[next])
             /*auto*/{
                 rnd_decisions++;
@@ -492,7 +492,7 @@
                         }/*auto*/
                     }/*auto*/
             }
-            seen[x] = 0;
+            analyze_stack.pop();
         }
     }
 
@@ -743,7 +743,7 @@
             }
 
             varDecayActivity();
-            claDecayActivity();
+            double  extra_lim = cla_inc / learnts.size();
 
             if (--learntsize_adjust_cnt == 0){
                 learntsize_adjust_confl *= learntsize_adjust_inc;
@@ -879,7 +879,7 @@
         printf("============================[ Search Statistics ]==============================\n");
         printf("| Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
         printf("|           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
-        printf("===============================================================================\n");
+        
     }
 
     // Search:
