--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1821,7 +1821,7 @@
                 this.slistener.end(Lbool.FALSE);
                 cancelUntil(0);
                 cancelLearntLiterals(learnedLiteralsLimit);
-                return false;
+                
             }
         }
         this.rootLevel = decisionLevel();
@@ -2556,7 +2556,7 @@
                 subset.add(q);
                 max = level;
             } else if (level == max) {
-                subset.add(q);
+                this.trail.pop();
             }
         }
         return subset;
