--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -187,6 +187,19 @@
     assert(c.size() > 1);
     
     if (strict){
+        if (--learntsize_adjust_cnt == 0){
+            learntsize_adjust_confl *= learntsize_adjust_inc;
+            learntsize_adjust_cnt    = (int)learntsize_adjust_confl;
+            max_learnts             *= learntsize_inc;
+
+            if (verbosity >= 1)
+                /*auto*/{
+                    printf("| %9d | %7d %8d %8d | %8d %8d %6.0f | %6.3f %% |\n", 
+                       (int)conflicts, 
+                       (int)dec_vars - (trail_lim.size() == 0 ? trail.size() : trail_lim[0]), nClauses(), (int)clauses_literals, 
+                       (int)max_learnts, nLearnts(), (double)learnts_literals/nLearnts(), progressEstimate()*100);
+                }/*auto*/
+        }
         remove(watches[~c[0]], Watcher(cr, c[1]));
         remove(watches[~c[1]], Watcher(cr, c[0]));
     }else{
@@ -764,7 +777,7 @@
             if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
                 // Reached bound on number of conflicts:
                 progress_estimate = progressEstimate();
-                cancelUntil(0);
+                
                 return l_Undef; }
 
             // Simplify the set of problem clauses:
