--- before: minisat_simplified.params
+++ after: minisat_simplified.params
@@ -1,12 +1,12 @@
-luby := 'True'
+luby := 'False'
 rnd-init := 'False'
 gc-frac := 0.2
-rinc := 2.0
-var-decay := 0.95
-cla-decay := 0.999
+rinc := 8746.228844570136
+var-decay := 0.9286001056838799
+cla-decay := 0.31139447014106203
 rnd-freq := 0.0
 rnd-seed := 91648253
-phase-saving := 2
+phase-saving := 1
 ccmin-mode := 2
 rfirst := 100
 pre := 'True'
@@ -17,4 +17,4 @@
 simp-gc-frac := 0.5
 sub-lim := 1000
 cl-lim := 20
-grow := 0
+grow := -9972
