--- before: minisat_simplified.params
+++ after: minisat_simplified.params
@@ -1,6 +1,6 @@
 luby := 'True'
 rnd-init := 'False'
-gc-frac := 0.2
+gc-frac := 8518.936317863527
 rinc := 2.0
 var-decay := 0.95
 cla-decay := 0.999
@@ -8,13 +8,13 @@
 rnd-seed := 91648253
 phase-saving := 2
 ccmin-mode := 2
-rfirst := 100
+rfirst := 28162
 pre := 'True'
 verb := '1'
 rcheck := 'False'
 asymm := 'False'
 elim := 'True'
-simp-gc-frac := 0.5
-sub-lim := 1000
+simp-gc-frac := 22914823.945328254
+sub-lim := 32018
 cl-lim := 20
-grow := 0
+grow := -875
