--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -373,7 +373,7 @@
                 core_added++;
             }else/*auto*/{
                 
-                claBumpActivity(c);
+                
             }/*auto*/
         }
 
@@ -1028,9 +1028,7 @@
     while (status == l_Undef){
         double rest_base = luby_restart ? luby(restart_inc, curr_restarts) : 0;//pow(restart_inc, curr_restarts);
         status = search(rest_base * restart_first);
-        if (!withinBudget())/*auto*/{
-             break;
-        }/*auto*/
+        
         curr_restarts++;
     }
 
