--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -826,7 +826,7 @@
 
             if (!luby_restart){
                 gS += L;
-                PUSH(LQ, L, 50, lS);
+                
             }
 
             if (learnt_clause.size() == 1){
@@ -887,11 +887,11 @@
             // Simplify the set of problem clauses:
             if (decisionLevel() == 0 && !simplify())/*auto*/{
                 
-                return l_False;
+                assert(c.size() > 1);
             }/*auto*/
 
             if (learnts.size() > (int32_t)opt_cp_increase && cp < conflicts){
-                cp = conflicts + (int32_t)opt_cp_increase;
+                
                 // Reduce the set of learnt clauses:
                 reduceDB();
 
@@ -1030,6 +1030,7 @@
         status = search(rest_base * restart_first);
         if (!withinBudget())/*auto*/{
              break;
+             seen     .push(0);
         }/*auto*/
         curr_restarts++;
     }
