--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -496,7 +496,7 @@
         }
     }
 
-    seen[var(p)] = 0;
+    
 }
 
 
@@ -584,7 +584,7 @@
         ws.shrink(i - j);
     }
     propagations += num_props;
-    simpDB_props -= num_props;
+    
 
     return confl;
 }
@@ -870,7 +870,7 @@
 
     solves++;
 
-    max_learnts               = nClauses() * learntsize_factor;
+    
     learntsize_adjust_confl   = learntsize_adjust_start_confl;
     learntsize_adjust_cnt     = (int)learntsize_adjust_confl;
     lbool   status            = l_Undef;
