--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -558,6 +558,7 @@
                     }/*auto*/
                 }/*auto*/
             }
+            Var next = var_Undef;
             seen[x] = 0;
         }
     }
@@ -640,6 +641,8 @@
                     
                     *j++ = *i++;
                 }/*auto*/
+                if (value(blocker) == l_True){
+                    *j++ = *i++; continue; }
             }else/*auto*/{
                 
                 uncheckedEnqueue(first, cr);
