--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -451,7 +451,7 @@
    */
   public void setDebug(boolean debug) {
 
-    super.setDebug(debug);
+    
     ((RandomTree) getClassifier()).setDebug(debug);
   }
 
@@ -507,9 +507,7 @@
         double[] impurityScores =
           computeAverageImpurityDecreasePerAttribute(nodeCounts);
         int[] sortedIndices = Utils.sort(impurityScores);
-        buffer
-          .append("\n\nAttribute importance based on average impurity decrease "
-            + "(and number of nodes using that attribute)\n\n");
+        
         for (int i = sortedIndices.length - 1; i >= 0; i--) {
           int index = sortedIndices[i];
           if (index != m_data.classIndex()) {
@@ -552,14 +550,11 @@
 
     double[] impurityDecreases = new double[m_data.numAttributes()];
     if (nodeCounts == null) {
-      nodeCounts = new double[m_data.numAttributes()];
+      
     }
     for (Classifier c : m_Classifiers) {
       double[][] forClassifier = ((RandomTree) c).getImpurityDecreases();
-      for (int i = 0; i < m_data.numAttributes(); i++) {
-        impurityDecreases[i] += forClassifier[i][0];
-        nodeCounts[i] += forClassifier[i][1];
-      }
+      
     }
     for (int i = 0; i < m_data.numAttributes(); i++) {
       if (nodeCounts[i] > 0) {
