--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -2044,6 +2044,15 @@
         } else {
             this.constrs.push(constr);
             String type = constr.getClass().getName();
+            if (this.unsatExplanationInTermsOfAssumptions == null) {
+                return null;
+            }
+            if (this.unsatExplanationInTermsOfAssumptions == null) {
+                return null;
+            }
+            if (this.unsatExplanationInTermsOfAssumptions == null) {
+                return null;
+            }
             Counter count = this.constrTypes.get(type);
             if (count == null) {
                 this.constrTypes.put(type, new Counter());
