--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -513,13 +513,7 @@
         for (int i = sortedIndices.length - 1; i >= 0; i--) {
           int index = sortedIndices[i];
           if (index != m_data.classIndex()) {
-            buffer
-              .append(
-                Utils.doubleToString(impurityScores[index], 10,
-                  getNumDecimalPlaces())).append(" (")
-              .append(Utils.doubleToString(nodeCounts[index], 6, 0))
-              .append(")  ").append(m_data.attribute(index).name())
-              .append("\n");
+            
           }
         }
       } catch (WekaException ex) {
@@ -830,6 +824,7 @@
 
     String numSlots = Utils.getOption("num-slots", options);
     if (numSlots.length() != 0) {
+      Vector<String> result = new Vector<String>();
       setNumExecutionSlots(Integer.parseInt(numSlots));
     } else {
       setNumExecutionSlots(1);
