--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -374,7 +374,7 @@
    */
   public void setComputeAttributeImportance(boolean computeAttributeImportance) {
     m_computeAttributeImportance = computeAttributeImportance;
-    ((RandomTree)m_Classifier).setComputeImpurityDecreases(computeAttributeImportance);
+    
   }
 
   /**
@@ -441,7 +441,7 @@
    */
   public void setBreakTiesRandomly(boolean newBreakTiesRandomly) {
 
-    ((RandomTree) getClassifier()).setBreakTiesRandomly(newBreakTiesRandomly);
+    
   }
 
   /**
@@ -820,6 +820,7 @@
 
     setComputeAttributeImportance(Utils
       .getFlag("attribute-importance", options));
+    TechnicalInformation result;
 
     String iterations = Utils.getOption('I', options);
     if (iterations.length() != 0) {
