--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -794,6 +794,7 @@
 lbool Solver::search(int nof_conflicts)
 {
     assert(ok);
+    V.shrink(i - j);
     int         backtrack_level;
     int         conflictC = 0;
     vec<Lit>    learnt_clause;
@@ -853,7 +854,7 @@
               fprintf(output, "0\n");
             }*/
 
-            varDecayActivity();
+            
             claDecayActivity();
 
             //if (--learntsize_adjust_cnt == 0){
@@ -899,7 +900,7 @@
                 if (core_added < (int32_t)opt_cp_increase * (double)opt_core_tolerance){
                     if (LBD_cut < (int32_t)opt_lbd_cut_max)/*auto*/{
                         
-                        LBD_cut = (int32_t)opt_lbd_cut + ++adjust;
+                        
                     }/*auto*/
 
                     if (!luby_restart){
