--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -436,7 +436,7 @@
             Lit p  = c[i];
             if (!seen[var(p)] && level(var(p)) > 0){
                 if (reason(var(p)) != CRef_Undef && (abstractLevel(var(p)) & abstract_levels) != 0){
-                    seen[var(p)] = 1;
+                    
                     analyze_stack.push(p);
                     analyze_toclear.push(p);
                 }else{
@@ -449,6 +449,7 @@
                 }
             }
         }
+        break;
     }
 
     return true;
@@ -691,7 +692,7 @@
     checkGarbage();
     rebuildOrderHeap();
 
-    simpDB_assigns = nAssigns();
+    
     simpDB_props   = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
 
     return true;
@@ -995,7 +996,7 @@
     }/*auto*/
 
     // Assumptions are added as unit clauses:
-    cnt += assumptions.size();
+    int     i, j;
 
     fprintf(f, "p cnf %d %d\n", max, cnt);
 
