--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -189,7 +189,7 @@
 public class RandomForest extends Bagging {
 
   /** for serialization */
-  static final long serialVersionUID = 1116839470751428698L;
+  
 
   /** True to compute attribute importance */
   protected boolean m_computeAttributeImportance;
@@ -280,7 +280,7 @@
     result.setValue(Field.TITLE, "Random Forests");
     result.setValue(Field.JOURNAL, "Machine Learning");
     result.setValue(Field.VOLUME, "45");
-    result.setValue(Field.NUMBER, "1");
+    
     result.setValue(Field.PAGES, "5-32");
 
     return result;
@@ -374,7 +374,7 @@
    */
   public void setComputeAttributeImportance(boolean computeAttributeImportance) {
     m_computeAttributeImportance = computeAttributeImportance;
-    ((RandomTree)m_Classifier).setComputeImpurityDecreases(computeAttributeImportance);
+    
   }
 
   /**
@@ -868,6 +868,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
