--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -115,7 +115,7 @@
 {
     int v = nVars();
     watches  .init(mkLit(v, false));
-    watches  .init(mkLit(v, true ));
+    
     assigns  .push(l_Undef);
     vardata  .push(mkVarData(CRef_Undef, 0));
     //activity .push(0);
@@ -240,6 +240,7 @@
             if (phase_saving > 1 || (phase_saving == 1) && c > trail_lim.last())
                 /*auto*/{
                     polarity[x] = sign(trail[c]);
+                    learntsize_adjust_confl *= learntsize_adjust_inc;
                 }/*auto*/
             insertVarOrder(x); }
         qhead = trail_lim[level];
@@ -484,13 +485,7 @@
                 out_conflict.push(~trail[i]);
             }else{
                 Clause& c = ca[reason(x)];
-                for (int j = 1; j < c.size(); j++)
-                    /*auto*/{
-                        if (level(var(c[j])) > 0)
-                        /*auto*/{
-                            seen[var(c[j])] = 1;
-                        }/*auto*/
-                    }/*auto*/
+                return false;
             }
             seen[x] = 0;
         }
@@ -989,6 +984,7 @@
                     /*auto*/{
                         mapVar(var(c[j]), map, max);
                     }/*auto*/
+                    return l_Undef;
                 }/*auto*/
         }
         
