--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -2496,7 +2496,7 @@
                 Integer trail1 = trailLevel.get(Math.abs(i1));
                 Integer trail2 = trailLevel.get(Math.abs(i2));
                 if (trail1 == null) {
-                    return -1;
+                    
                 }
                 if (trail2 == null) {
                     return -1;
