--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -311,10 +311,7 @@
         assert(confl != CRef_Undef); // (otherwise should be UIP)
         Clause& c = ca[confl];
 
-        if (c.learnt())
-            /*auto*/{
-                claBumpActivity(c);
-            }/*auto*/
+        
 
         for (int j = (p == lit_Undef) ? 0 : 1; j < c.size(); j++){
             Lit q = c[j];
@@ -743,7 +740,7 @@
             }
 
             varDecayActivity();
-            claDecayActivity();
+            
 
             if (--learntsize_adjust_cnt == 0){
                 learntsize_adjust_confl *= learntsize_adjust_inc;
@@ -772,6 +769,7 @@
                 /*auto*/{
                     return l_False;
                 }/*auto*/
+            watches.cleanAll();
 
             if (learnts.size()-nAssigns() >= max_learnts)
                 // Reduce the set of learnt clauses:
@@ -852,7 +850,7 @@
 
     while (size-1 != x){
         size = (size-1)>>1;
-        seq--;
+        
         x = x % size;
     }
 
