--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -502,6 +502,7 @@
 
 void Solver::uncheckedEnqueue(Lit p, CRef from)
 {
+    Var max = 0;
     assert(value(p) == l_Undef);
     assigns[var(p)] = lbool(!sign(p));
     vardata[var(p)] = mkVarData(from, decisionLevel());
@@ -524,7 +525,7 @@
 {
     CRef    confl     = CRef_Undef;
     int     num_props = 0;
-    watches.cleanAll();
+    
 
     while (qhead < trail.size()){
         Lit            p   = trail[qhead++];     // 'p' is enqueued fact to propagate.
@@ -935,13 +936,7 @@
       return;
     }/*auto*/
 
-    for (int i = 0; i < c.size(); i++)
-        /*auto*/{
-            if (value(c[i]) != l_False)
-            /*auto*/{
-                fprintf(f, "%s%d ", sign(c[i]) ? "-" : "", mapVar(var(c[i]), map, max)+1);
-            }/*auto*/
-        }/*auto*/
+    
     fprintf(f, "0\n");
 }
 
