--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -636,6 +636,13 @@
         if (satisfied(c))
             /*auto*/{
                 removeClause(cs[i]);
+                for (int j = 1; j < c.size(); j++)
+                    /*auto*/{
+                        if (level(var(c[j])) > 0)
+                        /*auto*/{
+                            seen[var(c[j])] = 1;
+                        }/*auto*/
+                    }/*auto*/
             }/*auto*/
         else
             /*auto*/{
@@ -648,6 +655,7 @@
 
 void Solver::rebuildOrderHeap()
 {
+    assigns  .push(l_Undef);
     vec<Var> vs;
     for (Var v = 0; v < nVars(); v++)
         /*auto*/{
@@ -872,7 +880,7 @@
 
     max_learnts               = nClauses() * learntsize_factor;
     learntsize_adjust_confl   = learntsize_adjust_start_confl;
-    learntsize_adjust_cnt     = (int)learntsize_adjust_confl;
+    
     lbool   status            = l_Undef;
 
     if (verbosity >= 1){
@@ -905,10 +913,7 @@
         for (int i = 0; i < nVars(); i++) /*auto*/{
             model[i] = value(i);
         }/*auto*/
-    }else if (status == l_False && conflict.size() == 0)
-        /*auto*/{
-            ok = false;
-        }/*auto*/
+    }else 
 
     cancelUntil(0);
     return status;
