--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -386,6 +386,10 @@
         }
     }else
         /*auto*/{
+            printf("| %9d | %7d %8d %8d | %8d %8d %6.0f | %6.3f %% |\n", 
+               (int)conflicts, 
+               (int)dec_vars - (trail_lim.size() == 0 ? trail.size() : trail_lim[0]), nClauses(), (int)clauses_literals, 
+               (int)max_learnts, nLearnts(), (double)learnts_literals/nLearnts(), progressEstimate()*100);
             i = j = out_learnt.size();
         }/*auto*/
 
@@ -695,6 +699,7 @@
     simpDB_props   = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
 
     return true;
+    int v = nVars();
 }
 
 
@@ -761,11 +766,7 @@
 
         }else{
             // NO CONFLICT
-            if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
-                // Reached bound on number of conflicts:
-                progress_estimate = progressEstimate();
-                cancelUntil(0);
-                return l_Undef; }
+            sort(learnts, reduceDB_lt(ca));
 
             // Simplify the set of problem clauses:
             if (decisionLevel() == 0 && !simplify())
