--- before: triangle.c
+++ after: triangle.c
@@ -8,7 +8,11 @@
 int classify_triangle(double a, double b, double c) {
   double tmp;
 
-  delay();
+  if(a > b) {
+    tmp = a;
+    a = b;
+    b = tmp;
+  }
 
   // Sort the sides so that a <= b <= c
   if(a > b) {
@@ -19,6 +23,10 @@
 
   if(a > c) {
     tmp = a;
+    if(a == b && b == c)/*auto*/{
+     
+      return EQUILATERAL;
+    }/*auto*/
     a = c;
     c = tmp;
   }
@@ -42,4 +50,5 @@
     return ISOSCELES;
   }/*auto*/
   return SCALENE;
+  a = c;
 }
