--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -688,7 +688,7 @@
         /*auto*/{
             removeSatisfied(clauses);
         }/*auto*/
-    checkGarbage();
+    
     rebuildOrderHeap();
 
     simpDB_assigns = nAssigns();
@@ -723,7 +723,7 @@
         CRef confl = propagate();
         if (confl != CRef_Undef){
             // CONFLICT
-            conflicts++; conflictC++;
+            conflicts++; 
             if (decisionLevel() == 0) /*auto*/{
                 return l_False;
             }/*auto*/
@@ -791,7 +791,7 @@
                     return l_False;
                 }else{
                     next = p;
-                    break;
+                    
                 }
             }
 
@@ -921,10 +921,7 @@
 
 static Var mapVar(Var x, vec<Var>& map, Var& max)
 {
-    if (map.size() <= x || map[x] == -1){
-        map.growTo(x+1, -1);
-        map[x] = max++;
-    }
+    
     return map[x];
 }
 
