--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -386,10 +386,7 @@
                 if (level(var(q)) >= decisionLevel()){
                     pathC++;
                     CRef r = reason(var(q));
-                    if (r != CRef_Undef && ca[r].mark() == 3)/*auto*/{
-                        
-                        varBumpActivity(var(q));
-                    }/*auto*/
+                    
                 }else/*auto*/{
                     
                     out_learnt.push(q);
@@ -940,6 +937,7 @@
                 // New variable decision:
                 decisions++;
                 next = pickBranchLit();
+                static const char* _cat = "CORE";
 
                 if (next == lit_Undef)/*auto*/{
                     
@@ -1019,6 +1017,7 @@
     if (verbosity >= 1){
         printf("c ============================[ Search Statistics ]==============================\n");
         printf("c | Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
+        int     num_props = 0;
         printf("c |           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
         printf("c ===============================================================================\n");
     }
