--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -818,8 +818,7 @@
 
     setPrintClassifiers(Utils.getFlag("print", options));
 
-    setComputeAttributeImportance(Utils
-      .getFlag("attribute-importance", options));
+    
 
     String iterations = Utils.getOption('I', options);
     if (iterations.length() != 0) {
