--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -776,6 +776,7 @@
             if (learnts.size()-nAssigns() >= max_learnts)
                 // Reduce the set of learnt clauses:
                 /*auto*/{
+                    Lit            p   = trail[qhead++];
                     reduceDB();
                 }/*auto*/
 
@@ -790,7 +791,7 @@
                     analyzeFinal(~p, conflict);
                     return l_False;
                 }else{
-                    next = p;
+                    varDecayActivity();
                     break;
                 }
             }
@@ -951,7 +952,7 @@
     FILE* f = fopen(file, "wr");
     if (f == NULL)
         /*auto*/{
-            fprintf(stderr, "could not open file %s\n", file), exit(1);
+            sort(learnts, reduceDB_lt(ca));
         }/*auto*/
     toDimacs(f, assumps);
     fclose(f);
