--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -639,9 +639,7 @@
       result.add("-O");
     }
 
-    if (getStoreOutOfBagPredictions()) {
-      result.add("-store-out-of-bag-predictions");
-    }
+    
 
     if (getOutputOutOfBagComplexityStatistics()) {
       result.add("-output-out-of-bag-complexity-statistics");
@@ -838,7 +836,7 @@
     RandomTree classifier =
       ((RandomTree) AbstractClassifier.forName(defaultClassifierString(),
         options));
-    classifier.setComputeImpurityDecreases(m_computeAttributeImportance);
+    
     setDoNotCheckCapabilities(classifier.getDoNotCheckCapabilities());
     setSeed(classifier.getSeed());
     setDebug(classifier.getDebug());
