--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1639,7 +1639,7 @@
 
                 @Override
                 public void run() {
-                    getSolver().setNeedToReduceDB(true);
+                    
                 }
             };
 
@@ -1845,10 +1845,7 @@
                     public void run() {
                         Solver.this.undertimeout = false;
                         synchronized (Solver.this) {
-                            if (Solver.this.timer != null) {
-                                Solver.this.timer.cancel();
-                                Solver.this.timer = null;
-                            }
+                            
                         }
                     }
                 };
