--- before: minisat_advanced.params
+++ after: minisat_advanced.params
@@ -1,9 +1,9 @@
 luby := 0.0
 rnd-init := 'False'
-gc-frac := 0.2
-rinc := 2.0
+gc-frac := 0.10549686991011387
+rinc := 9881.74997952102
 var-decay := 0.8
-lbd-cut := 5.0
+lbd-cut := 4.0291110014207
 lbd-cut-max := 10.0
 cla-decay := 0.999
 rnd-freq := 0.0
@@ -17,7 +17,7 @@
 rcheck := 'False'
 asymm := 'False'
 elim := 'True'
-simp-gc-frac := 0.5
+simp-gc-frac := 2866.5040878622635
 @sub-lim$flag := 'False'
 @cl-lim$flag := 'False'
 grow := 0
