--- before: minisat_simplified.params
+++ after: minisat_simplified.params
@@ -1,8 +1,8 @@
 luby := 'True'
 rnd-init := 'False'
-gc-frac := 0.2
+gc-frac := 15188.641475005024
 rinc := 2.0
-var-decay := 0.95
+var-decay := 0.7368851214228137
 cla-decay := 0.999
 rnd-freq := 0.0
 rnd-seed := 91648253
@@ -10,7 +10,7 @@
 ccmin-mode := 2
 rfirst := 100
 pre := 'True'
-verb := '1'
+verb := '2'
 rcheck := 'False'
 asymm := 'False'
 elim := 'True'
