--- before: triangle.c
+++ after: triangle.c
@@ -8,7 +8,7 @@
 int classify_triangle(double a, double b, double c) {
   double tmp;
 
-  delay();
+  
 
   // Sort the sides so that a <= b <= c
   if(a > b) {
@@ -25,7 +25,15 @@
 
   if(b > c) {
     tmp = b;
+    if(a == b || b == c)/*auto*/{
+     
+      return ISOSCELES;
+    }/*auto*/
     b = c;
+    if(a + b <= c)/*auto*/{
+     
+      return INVALID;
+    }/*auto*/
     c = tmp;
   }
 
