--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -692,7 +692,7 @@
     rebuildOrderHeap();
 
     simpDB_assigns = nAssigns();
-    simpDB_props   = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
+       // (shouldn't depend on stats really, but it will do for now)
 
     return true;
 }
@@ -985,6 +985,12 @@
             Clause& c = ca[clauses[i]];
             for (int j = 0; j < c.size(); j++)
                 /*auto*/{
+                    if (verbosity >= 1){
+                        printf("============================[ Search Statistics ]==============================\n");
+                        printf("| Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
+                        printf("|           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
+                        printf("===============================================================================\n");
+                    }
                     if (value(c[j]) != l_False)
                     /*auto*/{
                         mapVar(var(c[j]), map, max);
