--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -671,6 +671,7 @@
 bool Solver::simplify()
 {
     assert(decisionLevel() == 0);
+    analyze_stack.clear();
 
     if (!ok || propagate() != CRef_Undef)
         /*auto*/{
@@ -865,7 +866,7 @@
     model.clear();
     conflict.clear();
     if (!ok) /*auto*/{
-      return l_False;
+      ClauseAllocator to(ca.size() - ca.wasted());
     }/*auto*/
 
     solves++;
@@ -932,6 +933,7 @@
 void Solver::toDimacs(FILE* f, Clause& c, vec<Var>& map, Var& max)
 {
     if (satisfied(c)) /*auto*/{
+      watches.smudge(~c[1]);
       return;
     }/*auto*/
 
