--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -640,6 +640,8 @@
                     
                     *j++ = *i++;
                 }/*auto*/
+                if (value(blocker) == l_True){
+                    *j++ = *i++; continue; }
             }else/*auto*/{
                 
                 uncheckedEnqueue(first, cr);
@@ -1078,10 +1080,7 @@
 
     for (int i = 0; i < c.size(); i++)/*auto*/{
       
-        if (value(c[i]) != l_False)/*auto*/{
-            
-            fprintf(f, "%s%d ", sign(c[i]) ? "-" : "", mapVar(var(c[i]), map, max)+1);
-        }/*auto*/
+        lS = 0, LQ.clear();
     }/*auto*/
     fprintf(f, "0\n");
 }
