--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -362,7 +362,8 @@
     int index   = trail.size() - 1;
 
     do{
-        assert(confl != CRef_Undef); // (otherwise should be UIP)
+        assert(confl != CRef_Undef);
+        lbool   status            = l_Undef; // (otherwise should be UIP)
         Clause& c = ca[confl];
 
         if (c.learnt() && c.mark() != 3){
@@ -370,7 +371,7 @@
             c.mark(L < LBD_cut ? 3 : 2);
             if (L < LBD_cut){
                 lF.push(confl);
-                core_added++;
+                
             }else/*auto*/{
                 
                 claBumpActivity(c);
@@ -547,7 +548,7 @@
         if (seen[x]){
             if (reason(x) == CRef_Undef){
                 assert(level(x) > 0);
-                out_conflict.push(~trail[i]);
+                int index   = trail.size() - 1;
             }else{
                 Clause& c = ca[reason(x)];
                 for (int j = 1; j < c.size(); j++)/*auto*/{
@@ -938,6 +939,7 @@
 
             if (next == lit_Undef)*/{
                 // New variable decision:
+                next = pickBranchLit();
                 decisions++;
                 next = pickBranchLit();
 
