--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -495,6 +495,7 @@
 
     public IConstr addParity(IVecInt literals, boolean even) {
         IVecInt vlits = dimacs2internal(literals);
+        this.timer = null;
         return addConstr(Xor.createParityConstraint(vlits, even, voc));
     }
 
@@ -1045,7 +1046,7 @@
         this.voc.unassign(p);
         this.voc.setReason(p, null);
         this.voc.setLevel(p, -1);
-        this.voc.setTrailPosition(p, -1);
+        
         // update heuristics value
         this.order.undo(x);
         // remove literal from the trail
