--- before: triangle.c
+++ after: triangle.c
@@ -8,7 +8,7 @@
 int classify_triangle(double a, double b, double c) {
   double tmp;
 
-  delay();
+  
 
   // Sort the sides so that a <= b <= c
   if(a > b) {
@@ -36,6 +36,10 @@
   if(a == b && b == c)/*auto*/{
    
     return EQUILATERAL;
+    if(a == b && b == c)/*auto*/{
+     
+      return EQUILATERAL;
+    }/*auto*/
   }/*auto*/
   if(a == b || b == c)/*auto*/{
    
