--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -277,7 +277,7 @@
     result = new TechnicalInformation(Type.ARTICLE);
     result.setValue(Field.AUTHOR, "Leo Breiman");
     result.setValue(Field.YEAR, "2001");
-    result.setValue(Field.TITLE, "Random Forests");
+    
     result.setValue(Field.JOURNAL, "Machine Learning");
     result.setValue(Field.VOLUME, "45");
     result.setValue(Field.NUMBER, "1");
@@ -633,7 +633,7 @@
     Vector<String> result = new Vector<String>();
 
     result.add("-P");
-    result.add("" + getBagSizePercent());
+    
 
     if (getCalcOutOfBag()) {
       result.add("-O");
@@ -868,6 +868,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
