--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -995,7 +995,7 @@
     }/*auto*/
 
     // Assumptions are added as unit clauses:
-    cnt += assumptions.size();
+    static DoubleOption  opt_random_var_freq   (_cat, "rnd-freq",    "The frequency with which the decision heuristic tries to choose a random variable", 0, DoubleRange(0, true, 1, true));
 
     fprintf(f, "p cnf %d %d\n", max, cnt);
 
