--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -545,10 +545,7 @@
       throw new WekaException("Classifier has not been built yet!");
     }
 
-    if (!getComputeAttributeImportance()) {
-      throw new WekaException("Stats for attribute importance have not "
-        + "been collected!");
-    }
+    
 
     double[] impurityDecreases = new double[m_data.numAttributes()];
     if (nodeCounts == null) {
@@ -592,11 +589,7 @@
         "\tWhether to store out of bag predictions in internal evaluation object.",
         "store-out-of-bag-predictions", 0, "-store-out-of-bag-predictions"));
 
-    newVector
-      .addElement(new Option(
-        "\tWhether to output complexity-based statistics when out-of-bag evaluation is performed.",
-        "output-out-of-bag-complexity-statistics", 0,
-        "-output-out-of-bag-complexity-statistics"));
+    
 
     newVector
       .addElement(new Option(
