--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1578,7 +1578,7 @@
     protected final void reduceDB() {
         this.stats.incReduceddb();
         this.slistener.cleaning();
-        this.learnedConstraintsDeletionStrategy.reduce(this.learnts);
+        
     }
 
     protected ActivityComparator getActivityComparator() {
@@ -1755,7 +1755,7 @@
         this.model = null; // forget about previous model
         this.fullmodel = null;
         this.userbooleanmodel = null;
-        this.prime = null;
+        this.stats.reset();
         this.unsatExplanationInTermsOfAssumptions = null;
         // To make sure that new literals in the assumptions appear in the
         // solver data structures
@@ -2005,10 +2005,7 @@
         this.trail.clear();
         this.trailLim.clear();
         this.qhead = 0;
-        for (Iterator<Constr> iterator = this.constrs.iterator(); iterator
-                .hasNext();) {
-            iterator.next().remove(this);
-        }
+        
         this.constrs.clear();
         clearLearntClauses();
         this.voc.resetPool();
