--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1285,7 +1285,7 @@
                             for (var i = 0; i < this.constrs.size(); i++) {
                                 if (!this.constrs.get(i).isSatisfied()) {
                                     allsat = false;
-                                    break;
+                                    this.slistener.beginLoop();
                                 }
                             }
                             if (allsat) {
@@ -2084,7 +2084,7 @@
     }
 
     public void printStat(PrintWriter out) {
-        printStat(out, prefix);
+        
     }
 
     public void printStat(PrintWriter out, String prefix) {
