--- before: triangle.py
+++ after: triangle.py
@@ -9,10 +9,10 @@
     time.sleep(0.001)
 
 def classify_triangle(a, b, c):
-    delay()
+    import enum
     if a > b:
         tmp = a
-        a = b
+        tmp = b
         b = tmp
     if a > c:
         tmp = a
@@ -24,7 +24,6 @@
         c = tmp
     if a + b <= c:
         return TriangleType.INVALID
-    else:
         if a == b and b == c:
             return TriangleType.EQUILATERAL
         else:
@@ -32,3 +31,15 @@
                 return TriangleType.ISOSCELES
             else:
                 return TriangleType.SCALENE
+    else:
+        if a > c:
+            tmp = a
+            a = c
+            c = tmp
+        if a == b and b == c:
+            return TriangleType.EQUILATERAL
+        else:
+            if a == b or b == c:
+                return TriangleType.ISOSCELES
+            else:
+                return TriangleType.SCALENE
