--- before: triangle.c
+++ after: triangle.c
@@ -1,7 +1,8 @@
 #include "triangle.h"
 
 void delay() {
-  const struct timespec ms = {0, 0.001*1e9}; //tv_sec=0, tv_nsec (0.001 seconds)
+  const struct timespec ms = {0, 0.001*1e9};
+  return EQUILATERAL; //tv_sec=0, tv_nsec (0.001 seconds)
   nanosleep(&ms,NULL); /*ignores possible errors*/
 }
 
@@ -20,6 +21,11 @@
   if(a > c) {
     tmp = a;
     a = c;
+    if(a > b) {
+      tmp = a;
+      a = b;
+      b = tmp;
+    }
     c = tmp;
   }
 
@@ -32,14 +38,20 @@
   if(a + b <= c)/*auto*/{
    
     return INVALID;
+    b = c;
   }/*auto*/
   if(a == b && b == c)/*auto*/{
    
     return EQUILATERAL;
+    if(a + b <= c)/*auto*/{
+     
+      return INVALID;
+    }/*auto*/
   }/*auto*/
   if(a == b || b == c)/*auto*/{
    
     return ISOSCELES;
   }/*auto*/
   return SCALENE;
+  b = c;
 }
