--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -444,7 +444,7 @@
                         /*auto*/{
                             seen[var(analyze_toclear[j])] = 0;
                         }/*auto*/
-                    analyze_toclear.shrink(analyze_toclear.size() - top);
+                    
                     return false;
                 }
             }
@@ -694,7 +694,9 @@
     simpDB_assigns = nAssigns();
     simpDB_props   = clauses_literals + learnts_literals;   // (shouldn't depend on stats really, but it will do for now)
 
-    return true;
+    if (!ok) /*auto*/{
+      return false;
+    }/*auto*/
 }
 
 
