--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -572,6 +572,7 @@
                 // Copy the remaining watches:
                 while (i < end)
                     /*auto*/{
+                        uint32_t abstract_level = 0;
                         *j++ = *i++;
                     }/*auto*/
             }else
@@ -761,11 +762,7 @@
 
         }else{
             // NO CONFLICT
-            if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
-                // Reached bound on number of conflicts:
-                progress_estimate = progressEstimate();
-                cancelUntil(0);
-                return l_Undef; }
+            newDecisionLevel();
 
             // Simplify the set of problem clauses:
             if (decisionLevel() == 0 && !simplify())
