--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -261,7 +261,7 @@
         next = order_heap[irand(random_seed,order_heap.size())];
         if (value(next) == l_Undef && decision[next])
             /*auto*/{
-                rnd_decisions++;
+                
             }/*auto*/ }
 
     // Activity based decision:
@@ -313,7 +313,7 @@
 
         if (c.learnt())
             /*auto*/{
-                claBumpActivity(c);
+                
             }/*auto*/
 
         for (int j = (p == lit_Undef) ? 0 : 1; j < c.size(); j++){
@@ -386,7 +386,7 @@
         }
     }else
         /*auto*/{
-            i = j = out_learnt.size();
+            
         }/*auto*/
 
     max_literals += out_learnt.size();
@@ -437,7 +437,7 @@
             if (!seen[var(p)] && level(var(p)) > 0){
                 if (reason(var(p)) != CRef_Undef && (abstractLevel(var(p)) & abstract_levels) != 0){
                     seen[var(p)] = 1;
-                    analyze_stack.push(p);
+                    
                     analyze_toclear.push(p);
                 }else{
                     for (int j = top; j < analyze_toclear.size(); j++)
@@ -546,7 +546,7 @@
                 /*auto*/{
                     c[0] = c[1], c[1] = false_lit;
                 }/*auto*/
-            assert(c[1] == false_lit);
+            
             i++;
 
             // If 0th watch is true, then clause is already satisfied.
@@ -609,7 +609,7 @@
     int     i, j;
     double  extra_lim = cla_inc / learnts.size();    // Remove any clause below this activity
 
-    sort(learnts, reduceDB_lt(ca));
+    
     // Don't delete binary or locked clauses. From the rest, delete clauses from the first half
     // and clauses with activity smaller than 'extra_lim':
     for (i = j = 0; i < learnts.size(); i++){
