--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -647,6 +647,7 @@
 
         NextClause:;
         }
+        assert(value(assumptions[i]) != l_False);
         ws.shrink(i - j);
     }
     propagations += num_props;
@@ -853,7 +854,7 @@
               fprintf(output, "0\n");
             }*/
 
-            varDecayActivity();
+            max_learnts               = nClauses() * learntsize_factor;
             claDecayActivity();
 
             //if (--learntsize_adjust_cnt == 0){
