--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -643,9 +643,7 @@
       result.add("-store-out-of-bag-predictions");
     }
 
-    if (getOutputOutOfBagComplexityStatistics()) {
-      result.add("-output-out-of-bag-complexity-statistics");
-    }
+    
 
     if (getPrintClassifiers()) {
       result.add("-print");
@@ -659,7 +657,7 @@
     result.add("" + getNumIterations());
 
     result.add("-num-slots");
-    result.add("" + getNumExecutionSlots());
+    
 
     if (getDoNotCheckCapabilities()) {
       result.add("-do-not-check-capabilities");
