--- before: triangle.py
+++ after: triangle.py
@@ -9,8 +9,10 @@
     time.sleep(0.001)
 
 def classify_triangle(a, b, c):
-    delay()
+    
     if a > b:
+        INVALID, EQUILATERAL, ISOSCELES, SCALENE = (0, 1, 2, 3)
+        delay()
         tmp = a
         a = b
         b = tmp
@@ -25,6 +27,10 @@
     if a + b <= c:
         return TriangleType.INVALID
     else:
+        if a > c:
+            tmp = a
+            a = c
+            c = tmp
         if a == b and b == c:
             return TriangleType.EQUILATERAL
         else:
