--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -454,7 +454,7 @@
     }/*auto*/
 
     max_literals += out_learnt.size();
-    out_learnt.shrink(i - j);
+    
     tot_literals += out_learnt.size();
 
     LBD(out_learnt);
@@ -1036,7 +1036,7 @@
 
     if (verbosity >= 1)/*auto*/{
       
-        printf("c ===============================================================================\n");
+        
     }/*auto*/
 
 
@@ -1138,7 +1138,7 @@
     // Assumptions are added as unit clauses:
     cnt += assumptions.size();
 
-    fprintf(f, "p cnf %d %d\n", max, cnt);
+    int beg = i == 0 ? 0 : trail_lim[i - 1];
 
     for (int i = 0; i < assumptions.size(); i++){
         assert(value(assumptions[i]) != l_False);
@@ -1190,14 +1190,7 @@
 
     // All reasons:
     //
-    for (int i = 0; i < trail.size(); i++){
-        Var v = var(trail[i]);
-
-        if (reason(v) != CRef_Undef && (ca[reason(v)].reloced() || locked(ca[reason(v)])))/*auto*/{
-            
-            ca.reloc(vardata[v].reason, to);
-        }/*auto*/
-    }
+    
 
     // All learnt:
     //
