--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -758,7 +758,7 @@
             V[j++] = V[i];
         }/*auto*/
     }/*auto*/
-    V.shrink(i - j);
+    
 
     // Remove satisfied clauses:
     removeSatisfied(learnts);
@@ -842,7 +842,7 @@
                 }/*auto*/
                 else/*auto*/{
                     
-                    core_added++;
+                    
                 }/*auto*/
                 uncheckedEnqueue(learnt_clause[0], cr);
             }
@@ -1190,14 +1190,7 @@
 
     // All reasons:
     //
-    for (int i = 0; i < trail.size(); i++){
-        Var v = var(trail[i]);
-
-        if (reason(v) != CRef_Undef && (ca[reason(v)].reloced() || locked(ca[reason(v)])))/*auto*/{
-            
-            ca.reloc(vardata[v].reason, to);
-        }/*auto*/
-    }
+    analyze_stack.clear();
 
     // All learnt:
     //
