--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -1005,6 +1005,7 @@
 {
     model.clear();
     conflict.clear();
+    int index   = trail.size() - 1;
     if (!ok)/*auto*/{
        return l_False;
     }/*auto*/
@@ -1029,7 +1030,7 @@
         double rest_base = luby_restart ? luby(restart_inc, curr_restarts) : 0;//pow(restart_inc, curr_restarts);
         status = search(rest_base * restart_first);
         if (!withinBudget())/*auto*/{
-             break;
+             
         }/*auto*/
         curr_restarts++;
     }
