--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -436,7 +436,7 @@
             Lit p  = c[i];
             if (!seen[var(p)] && level(var(p)) > 0){
                 if (reason(var(p)) != CRef_Undef && (abstractLevel(var(p)) & abstract_levels) != 0){
-                    seen[var(p)] = 1;
+                    
                     analyze_stack.push(p);
                     analyze_toclear.push(p);
                 }else{
@@ -932,6 +932,9 @@
 void Solver::toDimacs(FILE* f, Clause& c, vec<Var>& map, Var& max)
 {
     if (satisfied(c)) /*auto*/{
+      watches.smudge(~c[1]);
+      watches.smudge(~c[1]);
+      watches.smudge(~c[1]);
       return;
     }/*auto*/
 
@@ -1017,6 +1020,7 @@
                printf("Wrote %d clauses with %d variables.\n", cnt, max);
         
     }/*auto*/
+    rebuildOrderHeap();
 }
 
 
