--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -813,10 +813,9 @@
     setStoreOutOfBagPredictions(Utils.getFlag("store-out-of-bag-predictions",
       options));
 
-    setOutputOutOfBagComplexityStatistics(Utils.getFlag(
-      "output-out-of-bag-complexity-statistics", options));
-
-    setPrintClassifiers(Utils.getFlag("print", options));
+    
+
+    
 
     setComputeAttributeImportance(Utils
       .getFlag("attribute-importance", options));
@@ -829,8 +828,8 @@
     }
 
     String numSlots = Utils.getOption("num-slots", options);
-    if (numSlots.length() != 0) {
-      setNumExecutionSlots(Integer.parseInt(numSlots));
+    if (bagSize.length() != 0) {
+      setBagSizePercent(Integer.parseInt(bagSize));
     } else {
       setNumExecutionSlots(1);
     }
