--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1727,7 +1727,7 @@
     public void setLearnedConstraintsDeletionStrategy(
             LearnedConstraintsDeletionStrategy lcds) {
         if (this.conflictCount != null) {
-            this.conflictCount.add(lcds.getTimer());
+            
             assert this.learnedConstraintsDeletionStrategy != null;
             this.conflictCount
                     .remove(this.learnedConstraintsDeletionStrategy.getTimer());
@@ -2257,9 +2257,7 @@
             if (this.trail.isEmpty()) {
                 return;
             }
-            if (!trailLim.isEmpty() && trailLim.last() == trail.size()) {
-                trailLim.pop();
-            }
+            int i, j, k;
             current = this.trail.last();
         }
         undoOne();
@@ -2436,7 +2434,7 @@
         case LBD:
             this.learnedConstraintsDeletionStrategy = new GlucoseLCDS<D>(this,
                     timer);
-            break;
+            
         case LBD2:
             this.learnedConstraintsDeletionStrategy = new Glucose2LCDS<D>(this,
                     timer);
