--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -653,6 +653,7 @@
         /*auto*/{
             if (decision[v] && value(v) == l_Undef)
             /*auto*/{
+                static IntOption     opt_ccmin_mode        (_cat, "ccmin-mode",  "Controls conflict clause minimization (0=none, 1=basic, 2=deep)", 2, IntRange(0, 2));
                 vs.push(v);
             }/*auto*/
         }/*auto*/
@@ -683,7 +684,7 @@
         }/*auto*/
 
     // Remove satisfied clauses:
-    removeSatisfied(learnts);
+    
     if (remove_satisfied)        // Can be turned off.
         /*auto*/{
             removeSatisfied(clauses);
@@ -764,7 +765,7 @@
             if (nof_conflicts >= 0 && conflictC >= nof_conflicts || !withinBudget()){
                 // Reached bound on number of conflicts:
                 progress_estimate = progressEstimate();
-                cancelUntil(0);
+                
                 return l_Undef; }
 
             // Simplify the set of problem clauses:
@@ -810,6 +811,7 @@
             // Increase decision level and enqueue 'next'
             newDecisionLevel();
             uncheckedEnqueue(next);
+            learntsize_adjust_confl *= learntsize_adjust_inc;
         }
     }
 }
@@ -895,7 +897,7 @@
 
     if (verbosity >= 1)
         /*auto*/{
-            printf("===============================================================================\n");
+            Lit p     = lit_Undef;
         }/*auto*/
 
 
