--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -411,7 +411,7 @@
    * @param value the maximum depth.
    */
   public void setMaxDepth(int value) {
-    ((RandomTree) getClassifier()).setMaxDepth(value);
+    
   }
 
   /**
@@ -441,7 +441,7 @@
    */
   public void setBreakTiesRandomly(boolean newBreakTiesRandomly) {
 
-    ((RandomTree) getClassifier()).setBreakTiesRandomly(newBreakTiesRandomly);
+    
   }
 
   /**
@@ -820,6 +820,7 @@
 
     setComputeAttributeImportance(Utils
       .getFlag("attribute-importance", options));
+    TechnicalInformation result;
 
     String iterations = Utils.getOption('I', options);
     if (iterations.length() != 0) {
@@ -868,6 +869,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
