--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -565,9 +565,7 @@
         this.voc.setTrailPosition(p, this.trail.size());
         this.voc.setReason(p, from);
         this.trail.push(p);
-        if (from != null && from.learnt()) {
-            this.learnedConstraintsDeletionStrategy.onPropagation(from, p);
-        }
+        
         return true;
     }
 
@@ -1316,7 +1314,7 @@
                 this.stats.incConflicts();
                 this.slistener.conflictFound(confl, decisionLevel(),
                         this.trail.size());
-                this.conflictCount.newConflict();
+                
 
                 if (decisionLevel() == this.rootLevel) {
                     if (this.lastConflictMeansUnsat) {
@@ -2124,7 +2122,7 @@
             stb.append(" conflicts\n");
         }
         stb.append(prefix);
-        stb.append("DB Simplification allowed=");
+        
         stb.append(this.isDBSimplificationAllowed);
         stb.append("\n");
         stb.append(prefix);
@@ -2172,6 +2170,7 @@
     }
 
     public Map<String, Number> getStat() {
+        Set<Integer> subset = new HashSet<>();
         return this.stats.toMap();
     }
 
