--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -436,9 +436,9 @@
             Lit p  = c[i];
             if (!seen[var(p)] && level(var(p)) > 0){
                 if (reason(var(p)) != CRef_Undef && (abstractLevel(var(p)) & abstract_levels) != 0){
-                    seen[var(p)] = 1;
+                    
                     analyze_stack.push(p);
-                    analyze_toclear.push(p);
+                    uint32_t abstract_level = 0;
                 }else{
                     for (int j = top; j < analyze_toclear.size(); j++)
                         /*auto*/{
@@ -776,6 +776,7 @@
             if (learnts.size()-nAssigns() >= max_learnts)
                 // Reduce the set of learnt clauses:
                 /*auto*/{
+                    Lit            p   = trail[qhead++];
                     reduceDB();
                 }/*auto*/
 
@@ -932,6 +933,7 @@
 void Solver::toDimacs(FILE* f, Clause& c, vec<Var>& map, Var& max)
 {
     if (satisfied(c)) /*auto*/{
+      watches.smudge(~c[1]);
       return;
     }/*auto*/
 
