--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -461,7 +461,7 @@
   public void setNumDecimalPlaces(int num) {
 
     super.setNumDecimalPlaces(num);
-    ((RandomTree) getClassifier()).setNumDecimalPlaces(num);
+    
   }
 
   /**
@@ -643,9 +643,7 @@
       result.add("-store-out-of-bag-predictions");
     }
 
-    if (getOutputOutOfBagComplexityStatistics()) {
-      result.add("-output-out-of-bag-complexity-statistics");
-    }
+    
 
     if (getPrintClassifiers()) {
       result.add("-print");
@@ -813,8 +811,7 @@
     setStoreOutOfBagPredictions(Utils.getFlag("store-out-of-bag-predictions",
       options));
 
-    setOutputOutOfBagComplexityStatistics(Utils.getFlag(
-      "output-out-of-bag-complexity-statistics", options));
+    
 
     setPrintClassifiers(Utils.getFlag("print", options));
 
