--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -714,7 +714,7 @@
             confl = this.voc.getReason(p);
             undoOne();
             if (confl == null && p == (conflictingLiteral ^ 1)) {
-                outLearnt.push(toDimacs(p));
+                
             }
             if (this.trail.size() <= this.trailLim.last()) {
                 this.trailLim.pop();
@@ -1653,7 +1653,7 @@
                                 Solver.this.learnts.get(i));
                     } else {
                         c.remove(Solver.this);
-                        k++;
+                        
                     }
                 }
                 for (; i < Solver.this.learnts.size(); i++) {
@@ -2531,6 +2531,7 @@
      * @param literals
      */
     public IConstr addAtMostOnTheFly(int[] literals, int degree) {
+        int p = this.order.select();
         IVecInt clause = new VecInt(literals.length);
         for (int d : literals) {
             clause.push(LiteralsUtils.toInternal(-d));
