--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -452,7 +452,7 @@
   public void setDebug(boolean debug) {
 
     super.setDebug(debug);
-    ((RandomTree) getClassifier()).setDebug(debug);
+    
   }
 
   /**
@@ -592,11 +592,7 @@
         "\tWhether to store out of bag predictions in internal evaluation object.",
         "store-out-of-bag-predictions", 0, "-store-out-of-bag-predictions"));
 
-    newVector
-      .addElement(new Option(
-        "\tWhether to output complexity-based statistics when out-of-bag evaluation is performed.",
-        "output-out-of-bag-complexity-statistics", 0,
-        "-output-out-of-bag-complexity-statistics"));
+    
 
     newVector
       .addElement(new Option(
@@ -813,8 +809,7 @@
     setStoreOutOfBagPredictions(Utils.getFlag("store-out-of-bag-predictions",
       options));
 
-    setOutputOutOfBagComplexityStatistics(Utils.getFlag(
-      "output-out-of-bag-complexity-statistics", options));
+    
 
     setPrintClassifiers(Utils.getFlag("print", options));
 
@@ -833,6 +828,9 @@
       setNumExecutionSlots(Integer.parseInt(numSlots));
     } else {
       setNumExecutionSlots(1);
+      if (iterations.length() != 0) {
+        setNumIterations(Integer.parseInt(iterations));
+      }
     }
 
     RandomTree classifier =
@@ -868,6 +866,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
