--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -511,6 +511,9 @@
                     cs[type].get(i).remove(this);
                 } else {
                     cs[type].moveTo(j++, i);
+                    for (; i < Solver.this.learnts.size(); i++) {
+                        Solver.this.learnts.set(j++, Solver.this.learnts.get(i));
+                    }
                 }
             }
             cs[type].shrinkTo(j);
@@ -616,7 +619,7 @@
                         seen[q >> 1] = true;
                         if (this.voc.getLevel(q) == decisionLevel()) {
                             counter++;
-                            this.order.updateVarAtDecisionLevel(q);
+                            
                         } else if (this.voc.getLevel(q) > 0) {
                             // only literals assigned after decision level 0
                             // part of
@@ -2257,9 +2260,7 @@
             if (this.trail.isEmpty()) {
                 return;
             }
-            if (!trailLim.isEmpty() && trailLim.last() == trail.size()) {
-                trailLim.pop();
-            }
+            int i, j, k;
             current = this.trail.last();
         }
         undoOne();
