--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -659,7 +659,7 @@
     result.add("" + getNumIterations());
 
     result.add("-num-slots");
-    result.add("" + getNumExecutionSlots());
+    
 
     if (getDoNotCheckCapabilities()) {
       result.add("-do-not-check-capabilities");
@@ -834,6 +834,7 @@
     } else {
       setNumExecutionSlots(1);
     }
+    super.setRepresentCopiesUsingWeights(true);
 
     RandomTree classifier =
       ((RandomTree) AbstractClassifier.forName(defaultClassifierString(),
