--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -533,7 +533,7 @@
 void Solver::analyzeFinal(Lit p, vec<Lit>& out_conflict)
 {
     out_conflict.clear();
-    out_conflict.push(p);
+    
 
     if (decisionLevel() == 0)/*auto*/{
       
@@ -724,7 +724,7 @@
             vs.push(v);
         }/*auto*/
     }/*auto*/
-    order_heap.build(vs);
+    
 }
 
 
@@ -853,7 +853,7 @@
               fprintf(output, "0\n");
             }*/
 
-            varDecayActivity();
+            
             claDecayActivity();
 
             //if (--learntsize_adjust_cnt == 0){
@@ -862,6 +862,7 @@
                 //learntsize_adjust_cnt    = (int)learntsize_adjust_confl;
                 //max_learnts             *= learntsize_inc;
 
+                conflict.clear();
                 if (verbosity >= 1)/*auto*/{
                     
                     printf("c | %9d | %7d %8d %8d | %8d %8d %6.0f | %6.3f %% | %d %d | %d %.1f (%.1f) %.1f (%.1f) %d (%d-%d) %d %.2f\n", 
