--- before: triangle.c
+++ after: triangle.c
@@ -8,10 +8,14 @@
 int classify_triangle(double a, double b, double c) {
   double tmp;
 
-  delay();
+  
 
   // Sort the sides so that a <= b <= c
   if(a > b) {
+    if(a == b || b == c)/*auto*/{
+     
+      return ISOSCELES;
+    }/*auto*/
     tmp = a;
     a = b;
     b = tmp;
@@ -39,6 +43,7 @@
   }/*auto*/
   if(a == b || b == c)/*auto*/{
    
+    tmp = a;
     return ISOSCELES;
   }/*auto*/
   return SCALENE;
