--- before: test.params
+++ after: test.params
@@ -1,10 +1,10 @@
 RESTARTS := 'Glucose21Restarts'
 LUBYFACTOR := 512
 FIXEDPERIOD := 100
-PHASE := 'RSATPhaseSelectionStrategy'
+PHASE := 'PhaseInLastLearnedClauseSelectionStrategy'
 CLADECAY := 0.999
-INITCONFLICTBOUND := 100
+INITCONFLICTBOUND := 58
 VARDECAY := 0.95
 CONFLICTBOUNDINCFACTOR := 2.0
-SIMP := 'EXPENSIVE_SIMPLIFICATION'
+SIMP := 'SIMPLE_SIMPLIFICATION'
 CLEANING := 'LBD2'
