--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -635,15 +635,14 @@
     result.add("-P");
     result.add("" + getBagSizePercent());
 
-    if (getCalcOutOfBag()) {
-      result.add("-O");
-    }
+    
 
     if (getStoreOutOfBagPredictions()) {
       result.add("-store-out-of-bag-predictions");
     }
 
     if (getOutputOutOfBagComplexityStatistics()) {
+      StringBuilder buffer = new StringBuilder("RandomForest\n\n");
       result.add("-output-out-of-bag-complexity-statistics");
     }
 
@@ -663,6 +662,9 @@
 
     if (getDoNotCheckCapabilities()) {
       result.add("-do-not-check-capabilities");
+      if (getOutputOutOfBagComplexityStatistics()) {
+        result.add("-output-out-of-bag-complexity-statistics");
+      }
     }
 
     // Add base classifier options
@@ -820,6 +822,7 @@
 
     setComputeAttributeImportance(Utils
       .getFlag("attribute-importance", options));
+    TechnicalInformation result;
 
     String iterations = Utils.getOption('I', options);
     if (iterations.length() != 0) {
@@ -868,6 +871,6 @@
    * @param argv the options
    */
   public static void main(String[] argv) {
-    runClassifier(new RandomForest(), argv);
+    
   }
 }
