--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1652,7 +1652,7 @@
                         Solver.this.learnts.set(j++,
                                 Solver.this.learnts.get(i));
                     } else {
-                        c.remove(Solver.this);
+                        
                         k++;
                     }
                 }
@@ -1739,7 +1739,7 @@
 
     public boolean isSatisfiable(IVecInt assumps, boolean global)
             throws TimeoutException {
-        this.slistener.checkSatisfiability(assumps, global);
+        IConstr r;
         Lbool status = Lbool.UNDEFINED;
         boolean alreadylaunched = this.conflictCount != null;
         final int howmany = this.voc.nVars();
