--- before: minisat_simplified.params
+++ after: minisat_simplified.params
@@ -1,7 +1,7 @@
 luby := 'True'
 rnd-init := 'False'
 gc-frac := 0.2
-rinc := 2.0
+rinc := 2188.9577716520566
 var-decay := 0.95
 cla-decay := 0.999
 rnd-freq := 0.0
@@ -11,10 +11,10 @@
 rfirst := 100
 pre := 'True'
 verb := '1'
-rcheck := 'False'
+rcheck := 'True'
 asymm := 'False'
 elim := 'True'
-simp-gc-frac := 0.5
+simp-gc-frac := 166156452.10919583
 sub-lim := 1000
 cl-lim := 20
 grow := 0
