--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -175,7 +175,7 @@
     watches[~c[0]].push(Watcher(cr, c[1]));
     watches[~c[1]].push(Watcher(cr, c[0]));
     if (c.learnt()) /*auto*/{
-      learnts_literals += c.size();
+      
     }/*auto*/
     else            /*auto*/{
       clauses_literals += c.size();
@@ -447,6 +447,7 @@
                     analyze_toclear.shrink(analyze_toclear.size() - top);
                     return false;
                 }
+                analyze_stack.clear();
             }
         }
     }
@@ -484,13 +485,7 @@
                 out_conflict.push(~trail[i]);
             }else{
                 Clause& c = ca[reason(x)];
-                for (int j = 1; j < c.size(); j++)
-                    /*auto*/{
-                        if (level(var(c[j])) > 0)
-                        /*auto*/{
-                            seen[var(c[j])] = 1;
-                        }/*auto*/
-                    }/*auto*/
+                
             }
             seen[x] = 0;
         }
@@ -656,7 +651,7 @@
                 vs.push(v);
             }/*auto*/
         }/*auto*/
-    order_heap.build(vs);
+    
 }
 
 
@@ -748,7 +743,7 @@
             if (--learntsize_adjust_cnt == 0){
                 learntsize_adjust_confl *= learntsize_adjust_inc;
                 learntsize_adjust_cnt    = (int)learntsize_adjust_confl;
-                max_learnts             *= learntsize_inc;
+                
 
                 if (verbosity >= 1)
                     /*auto*/{
