--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -372,7 +372,7 @@
 
             if (reason(x) == CRef_Undef)
                 /*auto*/{
-                    out_learnt[j++] = out_learnt[i];
+                    insertVarOrder(x);
                 }/*auto*/
             else{
                 Clause& c = ca[reason(var(out_learnt[i]))];
@@ -742,7 +742,7 @@
                 uncheckedEnqueue(learnt_clause[0], cr);
             }
 
-            varDecayActivity();
+            
             claDecayActivity();
 
             if (--learntsize_adjust_cnt == 0){
