--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -598,10 +598,7 @@
         "output-out-of-bag-complexity-statistics", 0,
         "-output-out-of-bag-complexity-statistics"));
 
-    newVector
-      .addElement(new Option(
-        "\tPrint the individual classifiers in the output", "print", 0,
-        "-print"));
+    
 
     newVector.addElement(new Option(
       "\tCompute and output attribute importance (mean impurity decrease "
