--- before: minisat_advanced.params
+++ after: minisat_advanced.params
@@ -2,7 +2,7 @@
 rnd-init := 'False'
 gc-frac := 0.2
 rinc := 2.0
-var-decay := 0.8
+var-decay := 0.9514478675232287
 lbd-cut := 5.0
 lbd-cut-max := 10.0
 cla-decay := 0.999
@@ -10,10 +10,10 @@
 rnd-seed := 91648253
 phase-saving := 2
 ccmin-mode := 2
-rfirst := 100
-cp-increase := 15000
-pre := 'True'
-verb := '1'
+rfirst := 960
+cp-increase := 17129
+pre := 'False'
+verb := '0'
 rcheck := 'False'
 asymm := 'False'
 elim := 'True'
