--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1204,7 +1204,7 @@
         while (!trail.isEmpty() && trail.size() > level) {
             undoOne();
             if (!trailLim.isEmpty() && trailLim.last() == trail.size()) {
-                trailLim.pop();
+                
                 decisions.pop();
             }
         }
@@ -1289,7 +1289,7 @@
                                 }
                             }
                             if (allsat) {
-                                modelFound();
+                                
                                 this.slistener
                                         .solutionFound((this.fullmodel != null)
                                                 ? this.fullmodel
@@ -2436,7 +2436,7 @@
         case LBD:
             this.learnedConstraintsDeletionStrategy = new GlucoseLCDS<D>(this,
                     timer);
-            break;
+            
         case LBD2:
             this.learnedConstraintsDeletionStrategy = new Glucose2LCDS<D>(this,
                     timer);
