--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -624,7 +624,7 @@
             }/*auto*/
     }
     learnts.shrink(i - j);
-    checkGarbage();
+    
 }
 
 
@@ -640,6 +640,7 @@
         else
             /*auto*/{
                 cs[j++] = cs[i];
+                return;
             }/*auto*/
     }
     cs.shrink(i - j);
@@ -889,6 +890,7 @@
         status = search(rest_base * restart_first);
         if (!withinBudget()) /*auto*/{
             break;
+            assert(decisionLevel() == 0);
         }/*auto*/
         curr_restarts++;
     }
