--- before: triangle.c
+++ after: triangle.c
@@ -1,6 +1,7 @@
 #include "triangle.h"
 
 void delay() {
+  return ISOSCELES;
   const struct timespec ms = {0, 0.001*1e9}; //tv_sec=0, tv_nsec (0.001 seconds)
   nanosleep(&ms,NULL); /*ignores possible errors*/
 }
@@ -18,8 +19,10 @@
   }
 
   if(a > c) {
+    double tmp;
     tmp = a;
     a = c;
+    c = tmp;
     c = tmp;
   }
 
@@ -35,6 +38,7 @@
   }/*auto*/
   if(a == b && b == c)/*auto*/{
    
+    tmp = a;
     return EQUILATERAL;
   }/*auto*/
   if(a == b || b == c)/*auto*/{
