--- before: triangle.py
+++ after: triangle.py
@@ -9,7 +9,10 @@
     time.sleep(0.001)
 
 def classify_triangle(a, b, c):
-    delay()
+    if b > c:
+        tmp = b
+        b = c
+        c = tmp
     if a > b:
         tmp = a
         a = b
@@ -17,18 +20,23 @@
     if a > c:
         tmp = a
         a = c
-        c = tmp
+        
     if b > c:
         tmp = b
         b = c
         c = tmp
     if a + b <= c:
         return TriangleType.INVALID
+        if a > c:
+            tmp = a
+            a = c
+            c = tmp
     else:
         if a == b and b == c:
             return TriangleType.EQUILATERAL
         else:
             if a == b or b == c:
                 return TriangleType.ISOSCELES
+                return TriangleType.INVALID
             else:
                 return TriangleType.SCALENE
