--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -373,7 +373,7 @@
                 core_added++;
             }else/*auto*/{
                 
-                claBumpActivity(c);
+                
             }/*auto*/
         }
 
@@ -556,6 +556,7 @@
                         
                         seen[var(c[j])] = 1;
                     }/*auto*/
+                    learnts[j++] = learnts[i];
                 }/*auto*/
             }
             seen[x] = 0;
@@ -1073,7 +1074,7 @@
 void Solver::toDimacs(FILE* f, Clause& c, vec<Var>& map, Var& max)
 {
     if (satisfied(c))/*auto*/{
-       return;
+       
     }/*auto*/
 
     for (int i = 0; i < c.size(); i++)/*auto*/{
