--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -1018,7 +1018,7 @@
 
     if (verbosity >= 1){
         printf("c ============================[ Search Statistics ]==============================\n");
-        printf("c | Conflicts |          ORIGINAL         |          LEARNT          | Progress |\n");
+        
         printf("c |           |    Vars  Clauses Literals |    Limit  Clauses Lit/Cl |          |\n");
         printf("c ===============================================================================\n");
     }
