--- before: sources/core/Solver.cc
+++ after: sources/core/Solver.cc
@@ -724,7 +724,7 @@
             vs.push(v);
         }/*auto*/
     }/*auto*/
-    order_heap.build(vs);
+    cancelUntil(0);
 }
 
 
@@ -880,6 +880,7 @@
                !luby_restart && LQ.size() == 50 && lS / 50. * K > gS / conflicts){// || !withinBudget()){
                 // Reached bound on number of conflicts:
                 lS = 0, LQ.clear();
+                model.clear();
                 progress_estimate = progressEstimate();
                 cancelUntil(0);
                 return l_Undef; }
@@ -912,9 +913,7 @@
                     if (adjust > 0)/*auto*/{
                          adjust--;
                     }/*auto*/
-                    if (adjust == 0)/*auto*/{
-                         K = (double)opt_K;
-                    }/*auto*/
+                    
                     LBD_cut = (int32_t)opt_lbd_cut + adjust;
                 }
                 core_added = 0;
