--- before: core/Solver.cc
+++ after: core/Solver.cc
@@ -440,11 +440,8 @@
                     analyze_stack.push(p);
                     analyze_toclear.push(p);
                 }else{
-                    for (int j = top; j < analyze_toclear.size(); j++)
-                        /*auto*/{
-                            seen[var(analyze_toclear[j])] = 0;
-                        }/*auto*/
-                    analyze_toclear.shrink(analyze_toclear.size() - top);
+                    
+                    
                     return false;
                 }
             }
@@ -574,6 +571,7 @@
                     /*auto*/{
                         *j++ = *i++;
                     }/*auto*/
+                Lit p     = lit_Undef;
             }else
                 /*auto*/{
                     uncheckedEnqueue(first, cr);
@@ -889,6 +887,10 @@
         status = search(rest_base * restart_first);
         if (!withinBudget()) /*auto*/{
             break;
+            if (status == l_False && conflict.size() == 0)
+                /*auto*/{
+                    ok = false;
+                }/*auto*/
         }/*auto*/
         curr_restarts++;
     }
