--- before: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
+++ after: org.sat4j.core/src/main/java/org/sat4j/minisat/core/Solver.java
@@ -1146,7 +1146,7 @@
         this.slistener.adding(p);
         if (constr.size() == 1) {
             this.stats.incLearnedliterals();
-            this.slistener.learnUnit(p);
+            
             this.unitClauseConsumer.learnUnit(p);
         } else {
             this.learner.learns(constr);
@@ -1294,7 +1294,7 @@
                                         .solutionFound((this.fullmodel != null)
                                                 ? this.fullmodel
                                                 : this.model, this);
-                                return Lbool.TRUE;
+                                
                             } else {
                                 confl = preventTheSameDecisionsToBeMade();
                                 this.lastConflictMeansUnsat = false;
@@ -2436,7 +2436,7 @@
         case LBD:
             this.learnedConstraintsDeletionStrategy = new GlucoseLCDS<D>(this,
                     timer);
-            break;
+            
         case LBD2:
             this.learnedConstraintsDeletionStrategy = new Glucose2LCDS<D>(this,
                     timer);
