--- before: src/main/java/weka/classifiers/trees/RandomForest.java
+++ after: src/main/java/weka/classifiers/trees/RandomForest.java
@@ -659,7 +659,7 @@
     result.add("" + getNumIterations());
 
     result.add("-num-slots");
-    result.add("" + getNumExecutionSlots());
+    
 
     if (getDoNotCheckCapabilities()) {
       result.add("-do-not-check-capabilities");
@@ -813,8 +813,7 @@
     setStoreOutOfBagPredictions(Utils.getFlag("store-out-of-bag-predictions",
       options));
 
-    setOutputOutOfBagComplexityStatistics(Utils.getFlag(
-      "output-out-of-bag-complexity-statistics", options));
+    
 
     setPrintClassifiers(Utils.getFlag("print", options));
 
